{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/data/fastq/rnaseq\n"
     ]
    }
   ],
   "source": [
    "cd /mnt/data/fastq/rnaseq/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;34m293\u001b[0m/                        \u001b[01;31mSRR629568_pass_2.fastq.gz\u001b[0m\r\n",
      "\u001b[01;34mkhmer_countgraph\u001b[0m/           SRR629569_pass_1.fastq\r\n",
      "\u001b[01;34mkhmer_k32\u001b[0m/                  \u001b[01;31mSRR629569_pass_1.fastq.gz\u001b[0m\r\n",
      "metadata.csv                SRR629569_pass_2.fastq\r\n",
      "metadata.json               \u001b[01;31mSRR629569_pass_2.fastq.gz\u001b[0m\r\n",
      "\u001b[01;34mMG\u001b[0m/                         SRR629570_pass_1.fastq\r\n",
      "\u001b[01;34mOS\u001b[0m/                         \u001b[01;31mSRR629570_pass_1.fastq.gz\u001b[0m\r\n",
      "\u001b[01;31msratoolkit.tar.gz\u001b[0m           SRR629570_pass_2.fastq\r\n",
      "SRR4098609_pass_1.fastq     \u001b[01;31mSRR629570_pass_2.fastq.gz\u001b[0m\r\n",
      "\u001b[01;31mSRR4098609_pass_1.fastq.gz\u001b[0m  SRR629571_pass_1.fastq\r\n",
      "SRR4098609_pass_2.fastq     \u001b[01;31mSRR629571_pass_1.fastq.gz\u001b[0m\r\n",
      "\u001b[01;31mSRR4098609_pass_2.fastq.gz\u001b[0m  SRR629571_pass_2.fastq\r\n",
      "SRR4098610_pass_1.fastq     \u001b[01;31mSRR629571_pass_2.fastq.gz\u001b[0m\r\n",
      "\u001b[01;31mSRR4098610_pass_1.fastq.gz\u001b[0m  SRR629572_pass_1.fastq\r\n",
      "SRR4098610_pass_2.fastq     \u001b[01;31mSRR629572_pass_1.fastq.gz\u001b[0m\r\n",
      "\u001b[01;31mSRR4098610_pass_2.fastq.gz\u001b[0m  SRR629572_pass_2.fastq\r\n",
      "SRR629557_pass_1.fastq      \u001b[01;31mSRR629572_pass_2.fastq.gz\u001b[0m\r\n",
      "\u001b[01;31mSRR629557_pass_1.fastq.gz\u001b[0m   SRR629573_pass_1.fastq\r\n",
      "SRR629557_pass_2.fastq      \u001b[01;31mSRR629573_pass_1.fastq.gz\u001b[0m\r\n",
      "\u001b[01;31mSRR629557_pass_2.fastq.gz\u001b[0m   SRR629573_pass_2.fastq\r\n",
      "SRR629559_pass_1.fastq      \u001b[01;31mSRR629573_pass_2.fastq.gz\u001b[0m\r\n",
      "\u001b[01;31mSRR629559_pass_1.fastq.gz\u001b[0m   SRR629576_pass_1.fastq\r\n",
      "SRR629559_pass_2.fastq      \u001b[01;31mSRR629576_pass_1.fastq.gz\u001b[0m\r\n",
      "\u001b[01;31mSRR629559_pass_2.fastq.gz\u001b[0m   SRR629576_pass_2.fastq\r\n",
      "SRR629561_pass_1.fastq      \u001b[01;31mSRR629576_pass_2.fastq.gz\u001b[0m\r\n",
      "\u001b[01;31mSRR629561_pass_1.fastq.gz\u001b[0m   SRR629577_pass_1.fastq\r\n",
      "SRR629561_pass_2.fastq      \u001b[01;31mSRR629577_pass_1.fastq.gz\u001b[0m\r\n",
      "\u001b[01;31mSRR629561_pass_2.fastq.gz\u001b[0m   SRR629577_pass_2.fastq\r\n",
      "SRR629562_pass_1.fastq      \u001b[01;31mSRR629577_pass_2.fastq.gz\u001b[0m\r\n",
      "\u001b[01;31mSRR629562_pass_1.fastq.gz\u001b[0m   SRR629578_pass_1.fastq\r\n",
      "SRR629562_pass_2.fastq      \u001b[01;31mSRR629578_pass_1.fastq.gz\u001b[0m\r\n",
      "\u001b[01;31mSRR629562_pass_2.fastq.gz\u001b[0m   SRR629578_pass_2.fastq\r\n",
      "SRR629563_pass_1.fastq      \u001b[01;31mSRR629578_pass_2.fastq.gz\u001b[0m\r\n",
      "\u001b[01;31mSRR629563_pass_1.fastq.gz\u001b[0m   SRR629580_pass_1.fastq\r\n",
      "SRR629563_pass_2.fastq      \u001b[01;31mSRR629580_pass_1.fastq.gz\u001b[0m\r\n",
      "\u001b[01;31mSRR629563_pass_2.fastq.gz\u001b[0m   SRR629580_pass_2.fastq\r\n",
      "SRR629564_pass_1.fastq      \u001b[01;31mSRR629580_pass_2.fastq.gz\u001b[0m\r\n",
      "\u001b[01;31mSRR629564_pass_1.fastq.gz\u001b[0m   SRR629581_pass_1.fastq\r\n",
      "SRR629564_pass_2.fastq      \u001b[01;31mSRR629581_pass_1.fastq.gz\u001b[0m\r\n",
      "\u001b[01;31mSRR629564_pass_2.fastq.gz\u001b[0m   SRR629581_pass_2.fastq\r\n",
      "SRR629565_pass_1.fastq      \u001b[01;31mSRR629581_pass_2.fastq.gz\u001b[0m\r\n",
      "\u001b[01;31mSRR629565_pass_1.fastq.gz\u001b[0m   SRR629582_pass_1.fastq\r\n",
      "SRR629565_pass_2.fastq      \u001b[01;31mSRR629582_pass_1.fastq.gz\u001b[0m\r\n",
      "\u001b[01;31mSRR629565_pass_2.fastq.gz\u001b[0m   SRR629582_pass_2.fastq\r\n",
      "SRR629566_pass_1.fastq      \u001b[01;31mSRR629582_pass_2.fastq.gz\u001b[0m\r\n",
      "\u001b[01;31mSRR629566_pass_1.fastq.gz\u001b[0m   SRR629583_pass_1.fastq\r\n",
      "SRR629566_pass_2.fastq      \u001b[01;31mSRR629583_pass_1.fastq.gz\u001b[0m\r\n",
      "\u001b[01;31mSRR629566_pass_2.fastq.gz\u001b[0m   SRR629583_pass_2.fastq\r\n",
      "SRR629568_pass_1.fastq      \u001b[01;31mSRR629583_pass_2.fastq.gz\u001b[0m\r\n",
      "\u001b[01;31mSRR629568_pass_1.fastq.gz\u001b[0m   \u001b[01;34mtrinity_out_dir\u001b[0m/\r\n",
      "SRR629568_pass_2.fastq\r\n"
     ]
    }
   ],
   "source": [
    "ls\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>LIBRARY_LAYOUT</th>\n",
       "      <th>LIBRARY_NAME</th>\n",
       "      <th>LIBRARY_SELECTION</th>\n",
       "      <th>LIBRARY_SOURCE</th>\n",
       "      <th>LIBRARY_STRATEGY</th>\n",
       "      <th>acc</th>\n",
       "      <th>cluster_name</th>\n",
       "      <th>is_public</th>\n",
       "      <th>load_done</th>\n",
       "      <th>name</th>\n",
       "      <th>static_data_available</th>\n",
       "      <th>status</th>\n",
       "      <th>total_bases</th>\n",
       "      <th>total_spots</th>\n",
       "      <th>ver</th>\n",
       "      <th>replicate</th>\n",
       "      <th>name_v2</th>\n",
       "      <th>cell_line</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>{'PAIRED': ''}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unspecified</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>SRR629557</td>\n",
       "      <td>public</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>A-431, replicate a</td>\n",
       "      <td>True</td>\n",
       "      <td>public</td>\n",
       "      <td>5256416326</td>\n",
       "      <td>26021863</td>\n",
       "      <td>2</td>\n",
       "      <td>a</td>\n",
       "      <td>A-431, replicate a</td>\n",
       "      <td>A-431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>{'PAIRED': ''}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unspecified</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>SRR629559</td>\n",
       "      <td>public</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>A-431, replicates b</td>\n",
       "      <td>True</td>\n",
       "      <td>public</td>\n",
       "      <td>6429905026</td>\n",
       "      <td>31831213</td>\n",
       "      <td>2</td>\n",
       "      <td>b</td>\n",
       "      <td>A-431, replicates b</td>\n",
       "      <td>A-431</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>{'PAIRED': ''}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unspecified</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>SRR629561</td>\n",
       "      <td>public</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>U-251 MG, replicate a</td>\n",
       "      <td>True</td>\n",
       "      <td>public</td>\n",
       "      <td>5791489076</td>\n",
       "      <td>28670738</td>\n",
       "      <td>1</td>\n",
       "      <td>a</td>\n",
       "      <td>U-251 MG, replicate a</td>\n",
       "      <td>U-251 MG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>{'PAIRED': ''}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unspecified</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>SRR629562</td>\n",
       "      <td>public</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>U-251 MG, replicate b</td>\n",
       "      <td>True</td>\n",
       "      <td>public</td>\n",
       "      <td>2451533610</td>\n",
       "      <td>12136305</td>\n",
       "      <td>1</td>\n",
       "      <td>b</td>\n",
       "      <td>U-251 MG, replicate b</td>\n",
       "      <td>U-251 MG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>{'PAIRED': ''}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unspecified</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>SRR629563</td>\n",
       "      <td>public</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>U-2 OS, replicate a</td>\n",
       "      <td>True</td>\n",
       "      <td>public</td>\n",
       "      <td>2069977426</td>\n",
       "      <td>10247413</td>\n",
       "      <td>2</td>\n",
       "      <td>a</td>\n",
       "      <td>U-2 OS, replicate a</td>\n",
       "      <td>U-2 OS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   LIBRARY_LAYOUT LIBRARY_NAME LIBRARY_SELECTION  LIBRARY_SOURCE  \\\n",
       "0  {'PAIRED': ''}          NaN       unspecified  TRANSCRIPTOMIC   \n",
       "1  {'PAIRED': ''}          NaN       unspecified  TRANSCRIPTOMIC   \n",
       "2  {'PAIRED': ''}          NaN       unspecified  TRANSCRIPTOMIC   \n",
       "3  {'PAIRED': ''}          NaN       unspecified  TRANSCRIPTOMIC   \n",
       "4  {'PAIRED': ''}          NaN       unspecified  TRANSCRIPTOMIC   \n",
       "\n",
       "  LIBRARY_STRATEGY        acc cluster_name  is_public  load_done  \\\n",
       "0          RNA-Seq  SRR629557       public       True       True   \n",
       "1          RNA-Seq  SRR629559       public       True       True   \n",
       "2          RNA-Seq  SRR629561       public       True       True   \n",
       "3          RNA-Seq  SRR629562       public       True       True   \n",
       "4          RNA-Seq  SRR629563       public       True       True   \n",
       "\n",
       "                    name  static_data_available  status  total_bases  \\\n",
       "0     A-431, replicate a                   True  public   5256416326   \n",
       "1    A-431, replicates b                   True  public   6429905026   \n",
       "2  U-251 MG, replicate a                   True  public   5791489076   \n",
       "3  U-251 MG, replicate b                   True  public   2451533610   \n",
       "4    U-2 OS, replicate a                   True  public   2069977426   \n",
       "\n",
       "   total_spots  ver replicate                name_v2 cell_line  \n",
       "0     26021863    2         a     A-431, replicate a     A-431  \n",
       "1     31831213    2         b    A-431, replicates b     A-431  \n",
       "2     28670738    1         a  U-251 MG, replicate a  U-251 MG  \n",
       "3     12136305    1         b  U-251 MG, replicate b  U-251 MG  \n",
       "4     10247413    2         a    U-2 OS, replicate a    U-2 OS  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "\n",
    "metadata = pd.read_csv('metadata.csv')\n",
    "metadata.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>LIBRARY_LAYOUT</th>\n",
       "      <th>LIBRARY_NAME</th>\n",
       "      <th>LIBRARY_SELECTION</th>\n",
       "      <th>LIBRARY_SOURCE</th>\n",
       "      <th>LIBRARY_STRATEGY</th>\n",
       "      <th>acc</th>\n",
       "      <th>cluster_name</th>\n",
       "      <th>is_public</th>\n",
       "      <th>load_done</th>\n",
       "      <th>name</th>\n",
       "      <th>static_data_available</th>\n",
       "      <th>status</th>\n",
       "      <th>total_bases</th>\n",
       "      <th>total_spots</th>\n",
       "      <th>ver</th>\n",
       "      <th>replicate</th>\n",
       "      <th>name_v2</th>\n",
       "      <th>cell_line</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>{'PAIRED': ''}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unspecified</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>SRR629581</td>\n",
       "      <td>public</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>PC-3, replicate b</td>\n",
       "      <td>True</td>\n",
       "      <td>public</td>\n",
       "      <td>2414023422</td>\n",
       "      <td>11950611</td>\n",
       "      <td>1</td>\n",
       "      <td>b</td>\n",
       "      <td>PC-3, replicate b</td>\n",
       "      <td>PC-3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>{'PAIRED': ''}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unspecified</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>SRR629582</td>\n",
       "      <td>public</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>RT-4, replicate a</td>\n",
       "      <td>True</td>\n",
       "      <td>public</td>\n",
       "      <td>4375683802</td>\n",
       "      <td>21661801</td>\n",
       "      <td>1</td>\n",
       "      <td>a</td>\n",
       "      <td>RT-4, replicate a</td>\n",
       "      <td>RT-4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>{'PAIRED': ''}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unspecified</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>SRR629583</td>\n",
       "      <td>public</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>RT-4, replicate b</td>\n",
       "      <td>True</td>\n",
       "      <td>public</td>\n",
       "      <td>4499721700</td>\n",
       "      <td>22275850</td>\n",
       "      <td>1</td>\n",
       "      <td>b</td>\n",
       "      <td>RT-4, replicate b</td>\n",
       "      <td>RT-4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>{'PAIRED': ''}</td>\n",
       "      <td>SHSY5Y_a</td>\n",
       "      <td>unspecified</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>SRR4098609</td>\n",
       "      <td>public</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>RNA-seq of cell line SHSY5Y replicate a</td>\n",
       "      <td>True</td>\n",
       "      <td>public</td>\n",
       "      <td>7465997770</td>\n",
       "      <td>36960385</td>\n",
       "      <td>1</td>\n",
       "      <td>a</td>\n",
       "      <td>SHSY5Y replicate a</td>\n",
       "      <td>SHSY5Y</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>{'PAIRED': ''}</td>\n",
       "      <td>SHSY5Y_b</td>\n",
       "      <td>unspecified</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>SRR4098610</td>\n",
       "      <td>public</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>RNA-seq of cell line SHSY5Y replicate b</td>\n",
       "      <td>True</td>\n",
       "      <td>public</td>\n",
       "      <td>3592995008</td>\n",
       "      <td>17787104</td>\n",
       "      <td>1</td>\n",
       "      <td>b</td>\n",
       "      <td>SHSY5Y replicate b</td>\n",
       "      <td>SHSY5Y</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    LIBRARY_LAYOUT LIBRARY_NAME LIBRARY_SELECTION  LIBRARY_SOURCE  \\\n",
       "18  {'PAIRED': ''}          NaN       unspecified  TRANSCRIPTOMIC   \n",
       "19  {'PAIRED': ''}          NaN       unspecified  TRANSCRIPTOMIC   \n",
       "20  {'PAIRED': ''}          NaN       unspecified  TRANSCRIPTOMIC   \n",
       "21  {'PAIRED': ''}     SHSY5Y_a       unspecified  TRANSCRIPTOMIC   \n",
       "22  {'PAIRED': ''}     SHSY5Y_b       unspecified  TRANSCRIPTOMIC   \n",
       "\n",
       "   LIBRARY_STRATEGY         acc cluster_name  is_public  load_done  \\\n",
       "18          RNA-Seq   SRR629581       public       True       True   \n",
       "19          RNA-Seq   SRR629582       public       True       True   \n",
       "20          RNA-Seq   SRR629583       public       True       True   \n",
       "21          RNA-Seq  SRR4098609       public       True       True   \n",
       "22          RNA-Seq  SRR4098610       public       True       True   \n",
       "\n",
       "                                       name  static_data_available  status  \\\n",
       "18                        PC-3, replicate b                   True  public   \n",
       "19                        RT-4, replicate a                   True  public   \n",
       "20                        RT-4, replicate b                   True  public   \n",
       "21  RNA-seq of cell line SHSY5Y replicate a                   True  public   \n",
       "22  RNA-seq of cell line SHSY5Y replicate b                   True  public   \n",
       "\n",
       "    total_bases  total_spots  ver replicate             name_v2 cell_line  \n",
       "18   2414023422     11950611    1         b   PC-3, replicate b      PC-3  \n",
       "19   4375683802     21661801    1         a   RT-4, replicate a      RT-4  \n",
       "20   4499721700     22275850    1         b   RT-4, replicate b      RT-4  \n",
       "21   7465997770     36960385    1         a  SHSY5Y replicate a    SHSY5Y  \n",
       "22   3592995008     17787104    1         b  SHSY5Y replicate b    SHSY5Y  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>LIBRARY_LAYOUT</th>\n",
       "      <th>LIBRARY_NAME</th>\n",
       "      <th>LIBRARY_SELECTION</th>\n",
       "      <th>LIBRARY_SOURCE</th>\n",
       "      <th>LIBRARY_STRATEGY</th>\n",
       "      <th>acc</th>\n",
       "      <th>cluster_name</th>\n",
       "      <th>is_public</th>\n",
       "      <th>load_done</th>\n",
       "      <th>name</th>\n",
       "      <th>static_data_available</th>\n",
       "      <th>status</th>\n",
       "      <th>total_bases</th>\n",
       "      <th>total_spots</th>\n",
       "      <th>ver</th>\n",
       "      <th>replicate</th>\n",
       "      <th>name_v2</th>\n",
       "      <th>cell_line</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>{'PAIRED': ''}</td>\n",
       "      <td>NaN</td>\n",
       "      <td>unspecified</td>\n",
       "      <td>TRANSCRIPTOMIC</td>\n",
       "      <td>RNA-Seq</td>\n",
       "      <td>SRR629563</td>\n",
       "      <td>public</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>U-2 OS, replicate a</td>\n",
       "      <td>True</td>\n",
       "      <td>public</td>\n",
       "      <td>2069977426</td>\n",
       "      <td>10247413</td>\n",
       "      <td>2</td>\n",
       "      <td>a</td>\n",
       "      <td>U-2 OS, replicate a</td>\n",
       "      <td>U-2 OS</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   LIBRARY_LAYOUT LIBRARY_NAME LIBRARY_SELECTION  LIBRARY_SOURCE  \\\n",
       "4  {'PAIRED': ''}          NaN       unspecified  TRANSCRIPTOMIC   \n",
       "\n",
       "  LIBRARY_STRATEGY        acc cluster_name  is_public  load_done  \\\n",
       "4          RNA-Seq  SRR629563       public       True       True   \n",
       "\n",
       "                  name  static_data_available  status  total_bases  \\\n",
       "4  U-2 OS, replicate a                   True  public   2069977426   \n",
       "\n",
       "   total_spots  ver replicate              name_v2 cell_line  \n",
       "4     10247413    2         a  U-2 OS, replicate a    U-2 OS  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata.query('cell_line == \"U-2 OS\"')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Note: HEK 293 didn't work because the cell line has a space so the filename has a space :(. Need to rerun using the sanitized cell line name. Should skip the ones that are already done since we don't use `--force`."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "|| This is the script load-into-counting.py in khmer.\r\n",
      "|| You are running khmer version 2.1.1\r\n",
      "|| You are also using screed version 1.0\r\n",
      "||\r\n",
      "|| If you use this script in a publication, please cite EACH of the following:\r\n",
      "||\r\n",
      "||   * MR Crusoe et al., 2015. http://dx.doi.org/10.12688/f1000research.6924.1\r\n",
      "||   * Q Zhang et al., http://dx.doi.org/10.1371/journal.pone.0101271\r\n",
      "||   * A. Döring et al. http://dx.doi.org:80/10.1186/1471-2105-9-11\r\n",
      "||\r\n",
      "|| Please see http://khmer.readthedocs.io/en/latest/citations.html for details.\r\n",
      "\r\n",
      "usage: load-into-counting.py [--version] [--info] [-h] [-k KSIZE]\r\n",
      "                             [-U UNIQUE_KMERS] [--fp-rate FP_RATE]\r\n",
      "                             [-M MAX_MEMORY_USAGE] [--small-count]\r\n",
      "                             [-T THREADS] [-b] [-s FORMAT] [-f] [-q]\r\n",
      "                             output_countgraph_filename\r\n",
      "                             input_sequence_filename\r\n",
      "                             [input_sequence_filename ...]\r\n",
      "\r\n",
      "Build a k-mer countgraph from the given sequences.\r\n",
      "\r\n",
      "positional arguments:\r\n",
      "  output_countgraph_filename\r\n",
      "                        The name of the file to write the k-mer countgraph to.\r\n",
      "  input_sequence_filename\r\n",
      "                        The names of one or more FAST[AQ] input sequence\r\n",
      "                        files.\r\n",
      "\r\n",
      "optional arguments:\r\n",
      "  --version             show program's version number and exit\r\n",
      "  --info                print citation information\r\n",
      "  -h, --help            show this help message and exit\r\n",
      "  -k KSIZE, --ksize KSIZE\r\n",
      "                        k-mer size to use (default: 32)\r\n",
      "  -U UNIQUE_KMERS, --unique-kmers UNIQUE_KMERS\r\n",
      "                        approximate number of unique kmers in the input set\r\n",
      "                        (default: 0)\r\n",
      "  --fp-rate FP_RATE     Override the automatic FP rate setting for the current\r\n",
      "                        script (default: None)\r\n",
      "  -M MAX_MEMORY_USAGE, --max-memory-usage MAX_MEMORY_USAGE\r\n",
      "                        maximum amount of memory to use for data structure\r\n",
      "                        (default: None)\r\n",
      "  --small-count         Reduce memory usage by using a smaller counter for\r\n",
      "                        individual kmers. (default: False)\r\n",
      "  -T THREADS, --threads THREADS\r\n",
      "                        Number of simultaneous threads to execute (default: 1)\r\n",
      "  -b, --no-bigcount     The default behaviour is to count past 255 using\r\n",
      "                        bigcount. This flag turns bigcount off, limiting\r\n",
      "                        counts to 255. (default: True)\r\n",
      "  -s FORMAT, --summary-info FORMAT\r\n",
      "                        What format should the machine readable run summary be\r\n",
      "                        in? (`json` or `tsv`, disabled by default) (default:\r\n",
      "                        None)\r\n",
      "  -f, --force           Overwrite output file if it exists (default: False)\r\n",
      "  -q, --quiet\r\n",
      "\r\n",
      "Note: with `-b`/`--no-bigcount` the output will be the exact size of the k-mer\r\n",
      "countgraph and this script will use a constant amount of memory. In exchange\r\n",
      "k-mer counts will stop at 255. The memory usage of this script with `-b` will\r\n",
      "be about 1.15x the product of the `-x` and `-N` numbers.\r\n",
      "\r\n",
      "Example:\r\n",
      "\r\n",
      "    load-into-counting.py -k 20 -x 5e7 out data/100k-filtered.fa\r\n",
      "\r\n",
      "Multiple threads can be used to accelerate the process, if you have extra cores\r\n",
      "to spare.\r\n",
      "\r\n",
      "Example:\r\n",
      "\r\n",
      "    load-into-counting.py -k 20 -x 5e7 -T 4 out data/100k-filtered.fa\r\n"
     ]
    }
   ],
   "source": [
    "! load-into-counting.py --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A-431 (2, 18)\n",
      "\n",
      "|| This is the script load-into-counting.py in khmer.\n",
      "|| You are running khmer version 2.1.1\n",
      "|| You are also using screed version 1.0\n",
      "||\n",
      "|| If you use this script in a publication, please cite EACH of the following:\n",
      "||\n",
      "||   * MR Crusoe et al., 2015. http://dx.doi.org/10.12688/f1000research.6924.1\n",
      "||   * Q Zhang et al., http://dx.doi.org/10.1371/journal.pone.0101271\n",
      "||   * A. Döring et al. http://dx.doi.org:80/10.1186/1471-2105-9-11\n",
      "||\n",
      "|| Please see http://khmer.readthedocs.io/en/latest/citations.html for details.\n",
      "\n",
      "\n",
      "PARAMETERS:\n",
      " - kmer size =     32 \t\t(-k)\n",
      " - n tables =      4 \t\t(-N)\n",
      " - max tablesize = 1.5e+10 \t(-x)\n",
      "Estimated memory usage is 60.0 Gb (6e+10 bytes = 4 bytes x 1.5e+10 entries / 1 entries per byte)\n",
      "--------\n",
      "ERROR: Not enough free space on disk for saved graph files;\n",
      "       Need at least 15.0 GB more.\n",
      "       Table size: 15.0 GB\n",
      "       Free space: 0.0 GB\n",
      "NOTE: This can be overridden using the --force argument\n",
      "A-549 (2, 18)\n",
      "\n",
      "|| This is the script load-into-counting.py in khmer.\n",
      "|| You are running khmer version 2.1.1\n",
      "|| You are also using screed version 1.0\n",
      "||\n",
      "|| If you use this script in a publication, please cite EACH of the following:\n",
      "||\n",
      "||   * MR Crusoe et al., 2015. http://dx.doi.org/10.12688/f1000research.6924.1\n",
      "||   * Q Zhang et al., http://dx.doi.org/10.1371/journal.pone.0101271\n",
      "||   * A. Döring et al. http://dx.doi.org:80/10.1186/1471-2105-9-11\n",
      "||\n",
      "|| Please see http://khmer.readthedocs.io/en/latest/citations.html for details.\n",
      "\n",
      "\n",
      "PARAMETERS:\n",
      " - kmer size =     32 \t\t(-k)\n",
      " - n tables =      4 \t\t(-N)\n",
      " - max tablesize = 1.5e+10 \t(-x)\n",
      "Estimated memory usage is 60.0 Gb (6e+10 bytes = 4 bytes x 1.5e+10 entries / 1 entries per byte)\n",
      "--------\n",
      "ERROR: Not enough free space on disk for saved graph files;\n",
      "       Need at least 15.0 GB more.\n",
      "       Table size: 15.0 GB\n",
      "       Free space: 0.0 GB\n",
      "NOTE: This can be overridden using the --force argument\n",
      "CACO-2 (2, 18)\n",
      "\n",
      "|| This is the script load-into-counting.py in khmer.\n",
      "|| You are running khmer version 2.1.1\n",
      "|| You are also using screed version 1.0\n",
      "||\n",
      "|| If you use this script in a publication, please cite EACH of the following:\n",
      "||\n",
      "||   * MR Crusoe et al., 2015. http://dx.doi.org/10.12688/f1000research.6924.1\n",
      "||   * Q Zhang et al., http://dx.doi.org/10.1371/journal.pone.0101271\n",
      "||   * A. Döring et al. http://dx.doi.org:80/10.1186/1471-2105-9-11\n",
      "||\n",
      "|| Please see http://khmer.readthedocs.io/en/latest/citations.html for details.\n",
      "\n",
      "\n",
      "PARAMETERS:\n",
      " - kmer size =     32 \t\t(-k)\n",
      " - n tables =      4 \t\t(-N)\n",
      " - max tablesize = 1.5e+10 \t(-x)\n",
      "Estimated memory usage is 60.0 Gb (6e+10 bytes = 4 bytes x 1.5e+10 entries / 1 entries per byte)\n",
      "--------\n",
      "ERROR: Not enough free space on disk for saved graph files;\n",
      "       Need at least 15.0 GB more.\n",
      "       Table size: 15.0 GB\n",
      "       Free space: 0.0 GB\n",
      "NOTE: This can be overridden using the --force argument\n",
      "HEK 293 (2, 18)\n",
      "mkdir: cannot create directory ‘/mnt/data/khmer_k32/HEK_293’: No space left on device\n",
      "\n",
      "|| This is the script load-into-counting.py in khmer.\n",
      "|| You are running khmer version 2.1.1\n",
      "|| You are also using screed version 1.0\n",
      "||\n",
      "|| If you use this script in a publication, please cite EACH of the following:\n",
      "||\n",
      "||   * MR Crusoe et al., 2015. http://dx.doi.org/10.12688/f1000research.6924.1\n",
      "||   * Q Zhang et al., http://dx.doi.org/10.1371/journal.pone.0101271\n",
      "||   * A. Döring et al. http://dx.doi.org:80/10.1186/1471-2105-9-11\n",
      "||\n",
      "|| Please see http://khmer.readthedocs.io/en/latest/citations.html for details.\n",
      "\n",
      "\n",
      "PARAMETERS:\n",
      " - kmer size =     32 \t\t(-k)\n",
      " - n tables =      4 \t\t(-N)\n",
      " - max tablesize = 1.5e+10 \t(-x)\n",
      "Estimated memory usage is 60.0 Gb (6e+10 bytes = 4 bytes x 1.5e+10 entries / 1 entries per byte)\n",
      "--------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ubuntu/anaconda/bin/load-into-counting.py\", line 221, in <module>\n",
      "    main()\n",
      "  File \"/home/ubuntu/anaconda/bin/load-into-counting.py\", line 123, in main\n",
      "    args.force)\n",
      "  File \"/home/ubuntu/anaconda/lib/python3.6/site-packages/khmer/kfile.py\", line 161, in check_space_for_graph\n",
      "    target = os.statvfs(dir_path)\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '/mnt/data/khmer_k32/HEK_293'\n",
      "HeLa (2, 18)\n",
      "\n",
      "|| This is the script load-into-counting.py in khmer.\n",
      "|| You are running khmer version 2.1.1\n",
      "|| You are also using screed version 1.0\n",
      "||\n",
      "|| If you use this script in a publication, please cite EACH of the following:\n",
      "||\n",
      "||   * MR Crusoe et al., 2015. http://dx.doi.org/10.12688/f1000research.6924.1\n",
      "||   * Q Zhang et al., http://dx.doi.org/10.1371/journal.pone.0101271\n",
      "||   * A. Döring et al. http://dx.doi.org:80/10.1186/1471-2105-9-11\n",
      "||\n",
      "|| Please see http://khmer.readthedocs.io/en/latest/citations.html for details.\n",
      "\n",
      "\n",
      "PARAMETERS:\n",
      " - kmer size =     32 \t\t(-k)\n",
      " - n tables =      4 \t\t(-N)\n",
      " - max tablesize = 1.5e+10 \t(-x)\n",
      "Estimated memory usage is 60.0 Gb (6e+10 bytes = 4 bytes x 1.5e+10 entries / 1 entries per byte)\n",
      "--------\n",
      "ERROR: Not enough free space on disk for saved graph files;\n",
      "       Need at least 15.0 GB more.\n",
      "       Table size: 15.0 GB\n",
      "       Free space: 0.0 GB\n",
      "NOTE: This can be overridden using the --force argument\n",
      "Hep-G2 (2, 18)\n",
      "\n",
      "|| This is the script load-into-counting.py in khmer.\n",
      "|| You are running khmer version 2.1.1\n",
      "|| You are also using screed version 1.0\n",
      "||\n",
      "|| If you use this script in a publication, please cite EACH of the following:\n",
      "||\n",
      "||   * MR Crusoe et al., 2015. http://dx.doi.org/10.12688/f1000research.6924.1\n",
      "||   * Q Zhang et al., http://dx.doi.org/10.1371/journal.pone.0101271\n",
      "||   * A. Döring et al. http://dx.doi.org:80/10.1186/1471-2105-9-11\n",
      "||\n",
      "|| Please see http://khmer.readthedocs.io/en/latest/citations.html for details.\n",
      "\n",
      "\n",
      "PARAMETERS:\n",
      " - kmer size =     32 \t\t(-k)\n",
      " - n tables =      4 \t\t(-N)\n",
      " - max tablesize = 1.5e+10 \t(-x)\n",
      "Estimated memory usage is 60.0 Gb (6e+10 bytes = 4 bytes x 1.5e+10 entries / 1 entries per byte)\n",
      "--------\n",
      "ERROR: Not enough free space on disk for saved graph files;\n",
      "       Need at least 15.0 GB more.\n",
      "       Table size: 15.0 GB\n",
      "       Free space: 0.0 GB\n",
      "NOTE: This can be overridden using the --force argument\n",
      "MCF-7 (2, 18)\n",
      "\n",
      "|| This is the script load-into-counting.py in khmer.\n",
      "|| You are running khmer version 2.1.1\n",
      "|| You are also using screed version 1.0\n",
      "||\n",
      "|| If you use this script in a publication, please cite EACH of the following:\n",
      "||\n",
      "||   * MR Crusoe et al., 2015. http://dx.doi.org/10.12688/f1000research.6924.1\n",
      "||   * Q Zhang et al., http://dx.doi.org/10.1371/journal.pone.0101271\n",
      "||   * A. Döring et al. http://dx.doi.org:80/10.1186/1471-2105-9-11\n",
      "||\n",
      "|| Please see http://khmer.readthedocs.io/en/latest/citations.html for details.\n",
      "\n",
      "\n",
      "PARAMETERS:\n",
      " - kmer size =     32 \t\t(-k)\n",
      " - n tables =      4 \t\t(-N)\n",
      " - max tablesize = 1.5e+10 \t(-x)\n",
      "Estimated memory usage is 60.0 Gb (6e+10 bytes = 4 bytes x 1.5e+10 entries / 1 entries per byte)\n",
      "--------\n",
      "ERROR: Not enough free space on disk for saved graph files;\n",
      "       Need at least 15.0 GB more.\n",
      "       Table size: 15.0 GB\n",
      "       Free space: 0.0 GB\n",
      "NOTE: This can be overridden using the --force argument\n",
      "PC-3 (2, 18)\n",
      "\n",
      "|| This is the script load-into-counting.py in khmer.\n",
      "|| You are running khmer version 2.1.1\n",
      "|| You are also using screed version 1.0\n",
      "||\n",
      "|| If you use this script in a publication, please cite EACH of the following:\n",
      "||\n",
      "||   * MR Crusoe et al., 2015. http://dx.doi.org/10.12688/f1000research.6924.1\n",
      "||   * Q Zhang et al., http://dx.doi.org/10.1371/journal.pone.0101271\n",
      "||   * A. Döring et al. http://dx.doi.org:80/10.1186/1471-2105-9-11\n",
      "||\n",
      "|| Please see http://khmer.readthedocs.io/en/latest/citations.html for details.\n",
      "\n",
      "\n",
      "PARAMETERS:\n",
      " - kmer size =     32 \t\t(-k)\n",
      " - n tables =      4 \t\t(-N)\n",
      " - max tablesize = 1.5e+10 \t(-x)\n",
      "Estimated memory usage is 60.0 Gb (6e+10 bytes = 4 bytes x 1.5e+10 entries / 1 entries per byte)\n",
      "--------\n",
      "ERROR: Not enough free space on disk for saved graph files;\n",
      "       Need at least 15.0 GB more.\n",
      "       Table size: 15.0 GB\n",
      "       Free space: 0.0 GB\n",
      "NOTE: This can be overridden using the --force argument\n",
      "RT-4 (2, 18)\n",
      "\n",
      "|| This is the script load-into-counting.py in khmer.\n",
      "|| You are running khmer version 2.1.1\n",
      "|| You are also using screed version 1.0\n",
      "||\n",
      "|| If you use this script in a publication, please cite EACH of the following:\n",
      "||\n",
      "||   * MR Crusoe et al., 2015. http://dx.doi.org/10.12688/f1000research.6924.1\n",
      "||   * Q Zhang et al., http://dx.doi.org/10.1371/journal.pone.0101271\n",
      "||   * A. Döring et al. http://dx.doi.org:80/10.1186/1471-2105-9-11\n",
      "||\n",
      "|| Please see http://khmer.readthedocs.io/en/latest/citations.html for details.\n",
      "\n",
      "\n",
      "PARAMETERS:\n",
      " - kmer size =     32 \t\t(-k)\n",
      " - n tables =      4 \t\t(-N)\n",
      " - max tablesize = 1.5e+10 \t(-x)\n",
      "Estimated memory usage is 60.0 Gb (6e+10 bytes = 4 bytes x 1.5e+10 entries / 1 entries per byte)\n",
      "--------\n",
      "ERROR: Not enough free space on disk for saved graph files;\n",
      "       Need at least 15.0 GB more.\n",
      "       Table size: 15.0 GB\n",
      "       Free space: 0.0 GB\n",
      "NOTE: This can be overridden using the --force argument\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SHSY5Y (2, 18)\n",
      "\n",
      "|| This is the script load-into-counting.py in khmer.\n",
      "|| You are running khmer version 2.1.1\n",
      "|| You are also using screed version 1.0\n",
      "||\n",
      "|| If you use this script in a publication, please cite EACH of the following:\n",
      "||\n",
      "||   * MR Crusoe et al., 2015. http://dx.doi.org/10.12688/f1000research.6924.1\n",
      "||   * Q Zhang et al., http://dx.doi.org/10.1371/journal.pone.0101271\n",
      "||   * A. Döring et al. http://dx.doi.org:80/10.1186/1471-2105-9-11\n",
      "||\n",
      "|| Please see http://khmer.readthedocs.io/en/latest/citations.html for details.\n",
      "\n",
      "\n",
      "PARAMETERS:\n",
      " - kmer size =     32 \t\t(-k)\n",
      " - n tables =      4 \t\t(-N)\n",
      " - max tablesize = 1.5e+10 \t(-x)\n",
      "Estimated memory usage is 60.0 Gb (6e+10 bytes = 4 bytes x 1.5e+10 entries / 1 entries per byte)\n",
      "--------\n",
      "ERROR: Not enough free space on disk for saved graph files;\n",
      "       Need at least 15.0 GB more.\n",
      "       Table size: 15.0 GB\n",
      "       Free space: 0.0 GB\n",
      "NOTE: This can be overridden using the --force argument\n",
      "U-2 OS (1, 18)\n",
      "mkdir: cannot create directory ‘/mnt/data/khmer_k32/U-2_OS’: No space left on device\n",
      "\n",
      "|| This is the script load-into-counting.py in khmer.\n",
      "|| You are running khmer version 2.1.1\n",
      "|| You are also using screed version 1.0\n",
      "||\n",
      "|| If you use this script in a publication, please cite EACH of the following:\n",
      "||\n",
      "||   * MR Crusoe et al., 2015. http://dx.doi.org/10.12688/f1000research.6924.1\n",
      "||   * Q Zhang et al., http://dx.doi.org/10.1371/journal.pone.0101271\n",
      "||   * A. Döring et al. http://dx.doi.org:80/10.1186/1471-2105-9-11\n",
      "||\n",
      "|| Please see http://khmer.readthedocs.io/en/latest/citations.html for details.\n",
      "\n",
      "\n",
      "PARAMETERS:\n",
      " - kmer size =     32 \t\t(-k)\n",
      " - n tables =      4 \t\t(-N)\n",
      " - max tablesize = 1.5e+10 \t(-x)\n",
      "Estimated memory usage is 60.0 Gb (6e+10 bytes = 4 bytes x 1.5e+10 entries / 1 entries per byte)\n",
      "--------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ubuntu/anaconda/bin/load-into-counting.py\", line 221, in <module>\n",
      "    main()\n",
      "  File \"/home/ubuntu/anaconda/bin/load-into-counting.py\", line 123, in main\n",
      "    args.force)\n",
      "  File \"/home/ubuntu/anaconda/lib/python3.6/site-packages/khmer/kfile.py\", line 161, in check_space_for_graph\n",
      "    target = os.statvfs(dir_path)\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '/mnt/data/khmer_k32/U-2_OS'\n",
      "U-251 MG (2, 18)\n",
      "mkdir: cannot create directory ‘/mnt/data/khmer_k32/U-251_MG’: No space left on device\n",
      "\n",
      "|| This is the script load-into-counting.py in khmer.\n",
      "|| You are running khmer version 2.1.1\n",
      "|| You are also using screed version 1.0\n",
      "||\n",
      "|| If you use this script in a publication, please cite EACH of the following:\n",
      "||\n",
      "||   * MR Crusoe et al., 2015. http://dx.doi.org/10.12688/f1000research.6924.1\n",
      "||   * Q Zhang et al., http://dx.doi.org/10.1371/journal.pone.0101271\n",
      "||   * A. Döring et al. http://dx.doi.org:80/10.1186/1471-2105-9-11\n",
      "||\n",
      "|| Please see http://khmer.readthedocs.io/en/latest/citations.html for details.\n",
      "\n",
      "\n",
      "PARAMETERS:\n",
      " - kmer size =     32 \t\t(-k)\n",
      " - n tables =      4 \t\t(-N)\n",
      " - max tablesize = 1.5e+10 \t(-x)\n",
      "Estimated memory usage is 60.0 Gb (6e+10 bytes = 4 bytes x 1.5e+10 entries / 1 entries per byte)\n",
      "--------\n",
      "Traceback (most recent call last):\n",
      "  File \"/home/ubuntu/anaconda/bin/load-into-counting.py\", line 221, in <module>\n",
      "    main()\n",
      "  File \"/home/ubuntu/anaconda/bin/load-into-counting.py\", line 123, in main\n",
      "    args.force)\n",
      "  File \"/home/ubuntu/anaconda/lib/python3.6/site-packages/khmer/kfile.py\", line 161, in check_space_for_graph\n",
      "    target = os.statvfs(dir_path)\n",
      "FileNotFoundError: [Errno 2] No such file or directory: '/mnt/data/khmer_k32/U-251_MG'\n"
     ]
    }
   ],
   "source": [
    "for cell_line, df in metadata.groupby('cell_line'):\n",
    "    print(cell_line, df.shape,)\n",
    "    read1 = df.acc + \"_pass_1.fastq.gz\"\n",
    "    read2 = df.acc + \"_pass_2.fastq.gz\"\n",
    "    reads = ' '.join(read1) + ' ' + ' '.join(read2)\n",
    "    \n",
    "    # Sanitize the cell line name\n",
    "    cell_line = cell_line.replace(' ', '_')\n",
    "    output_folder = f\"/mnt/data/khmer_k32/{cell_line}\"\n",
    "    output_file = f'{output_folder}/khmer_countgraph'\n",
    "#     print(output_file)\n",
    "    ! mkdir -p $output_folder\n",
    "    # ksize of 32 is the default\n",
    "    ! load-into-counting.py --summary-info tsv --max-memory-usage 60G --ksize 32 --threads 16  $output_file $reads\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "range(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "ls -lh /mnt/data/khmer_k32/*/*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "2.3283064365386963e-10"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.power(4., -16)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "-0.02328306436809747"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "1e8*np.log(1-np.power(4., -16))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.97698589473394"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.exp(1e8*np.log(1-np.power(4., -16)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.02301410526605996"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "1- np.exp(1e8*np.log(1-np.power(4., -16)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "23.015800594086954"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "desired_probability = 0.01\n",
    "genome_size = 1e8\n",
    "\n",
    "np.log(genome_size*(1-desired_probability)/desired_probability)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "25.32743552260092"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "desired_probability = 0.001\n",
    "genome_size = 1e8\n",
    "\n",
    "np.log(genome_size*(1-desired_probability)/desired_probability)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "32.23619030191614"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "desired_probability = 1e-6\n",
    "genome_size = 1e8\n",
    "\n",
    "np.log(genome_size*(1-desired_probability)/desired_probability)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "|| This is the script extract-paired-reads.py in khmer.\r\n",
      "|| You are running khmer version 2.1.1\r\n",
      "|| You are also using screed version 1.0\r\n",
      "||\r\n",
      "|| If you use this script in a publication, please cite EACH of the following:\r\n",
      "||\r\n",
      "||   * MR Crusoe et al., 2015. http://dx.doi.org/10.12688/f1000research.6924.1\r\n",
      "||\r\n",
      "|| Please see http://khmer.readthedocs.io/en/latest/citations.html for details.\r\n",
      "\r\n",
      "usage: extract-paired-reads.py [--version] [--info] [-h] [-d OUTPUT_DIR]\r\n",
      "                               [-p filename] [-s filename] [-f]\r\n",
      "                               [--gzip | --bzip]\r\n",
      "                               [infile]\r\n",
      "\r\n",
      "Take a mixture of reads and split into pairs and orphans.\r\n",
      "\r\n",
      "positional arguments:\r\n",
      "  infile\r\n",
      "\r\n",
      "optional arguments:\r\n",
      "  --version             show program's version number and exit\r\n",
      "  --info                print citation information\r\n",
      "  -h, --help            show this help message and exit\r\n",
      "  -d OUTPUT_DIR, --output-dir OUTPUT_DIR\r\n",
      "                        Output split reads to specified directory. Creates\r\n",
      "                        directory if necessary (default: )\r\n",
      "  -p filename, --output-paired filename\r\n",
      "                        Output paired reads to this file (default: None)\r\n",
      "  -s filename, --output-single filename\r\n",
      "                        Output orphaned reads to this file (default: None)\r\n",
      "  -f, --force           Overwrite output file if it exists (default: False)\r\n",
      "  --gzip                Compress output using gzip (default: False)\r\n",
      "  --bzip                Compress output using bzip2 (default: False)\r\n",
      "\r\n",
      "Many read-handling programs (assemblers, mappers, etc.) require that you give\r\n",
      "them either perfectly interleaved files, or files containing only single reads.\r\n",
      "This script takes files that were originally interleaved but where reads may\r\n",
      "have been orphaned (via error filtering, application of abundance filtering,\r\n",
      "digital normalization in non-paired mode, or partitioning) and separates the\r\n",
      "interleaved reads from the orphaned reads.\r\n",
      "\r\n",
      "The default output is two files, `<input file>.pe` and `<input file>.se`,\r\n",
      "placed in the current directory. The .pe file contains interleaved and properly\r\n",
      "paired sequences, while the .se file contains orphan sequences.\r\n",
      "\r\n",
      "The directory into which the interleaved and orphaned reads are output may be\r\n",
      "specified using `-d`/`--output-dir`. This directory will be created if it does\r\n",
      "not already exist.\r\n",
      "\r\n",
      "Alternatively, you can specify the filenames directly with `-p`/`--output-\r\n",
      "paired` and `-s`/`--output-single`, which will override the `-d`/`--output-dir`\r\n",
      "option.\r\n",
      "\r\n",
      "Example:\r\n",
      "\r\n",
      "    extract-paired-reads.py tests/test-data/paired.fq\r\n"
     ]
    }
   ],
   "source": [
    "! extract-paired-reads.py --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
