{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\r\n",
      "\r\n",
      "\r\n",
      "###############################################################################\r\n",
      "#\r\n",
      "\r\n",
      "     ______  ____   ____  ____   ____  ______  __ __\r\n",
      "    |      ||    \\ |    ||    \\ |    ||      ||  |  |\r\n",
      "    |      ||  D  ) |  | |  _  | |  | |      ||  |  |\r\n",
      "    |_|  |_||    /  |  | |  |  | |  | |_|  |_||  ~  |\r\n",
      "      |  |  |    \\  |  | |  |  | |  |   |  |  |___, |\r\n",
      "      |  |  |  .  \\ |  | |  |  | |  |   |  |  |     |\r\n",
      "      |__|  |__|\\_||____||__|__||____|  |__|  |____/\r\n",
      "\r\n",
      "#\r\n",
      "#\r\n",
      "# Required:\r\n",
      "#\r\n",
      "#  --seqType <string>      :type of reads: ('fa' or 'fq')\r\n",
      "#\r\n",
      "#  --max_memory <string>      :suggested max memory to use by Trinity where limiting can be enabled. (jellyfish, sorting, etc)\r\n",
      "#                            provided in Gb of RAM, ie.  '--max_memory 10G'\r\n",
      "#\r\n",
      "#  If paired reads:\r\n",
      "#      --left  <string>    :left reads, one or more file names (separated by commas, no spaces)\r\n",
      "#      --right <string>    :right reads, one or more file names (separated by commas, no spaces)\r\n",
      "#\r\n",
      "#  Or, if unpaired reads:\r\n",
      "#      --single <string>   :single reads, one or more file names, comma-delimited (note, if single file contains pairs, can use flag: --run_as_paired )\r\n",
      "#\r\n",
      "#  Or,\r\n",
      "#      --samples_file <string>         tab-delimited text file indicating biological replicate relationships.\r\n",
      "#                                   ex.\r\n",
      "#                                        cond_A    cond_A_rep1    A_rep1_left.fq    A_rep1_right.fq\r\n",
      "#                                        cond_A    cond_A_rep2    A_rep2_left.fq    A_rep2_right.fq\r\n",
      "#                                        cond_B    cond_B_rep1    B_rep1_left.fq    B_rep1_right.fq\r\n",
      "#                                        cond_B    cond_B_rep2    B_rep2_left.fq    B_rep2_right.fq\r\n",
      "#\r\n",
      "#                      # if single-end instead of paired-end, then leave the 4th column above empty.\r\n",
      "#\r\n",
      "####################################\r\n",
      "##  Misc:  #########################\r\n",
      "#\r\n",
      "#  --SS_lib_type <string>          :Strand-specific RNA-Seq read orientation.\r\n",
      "#                                   if paired: RF or FR,\r\n",
      "#                                   if single: F or R.   (dUTP method = RF)\r\n",
      "#                                   See web documentation.\r\n",
      "#\r\n",
      "#  --CPU <int>                     :number of CPUs to use, default: 2\r\n",
      "#  --min_contig_length <int>       :minimum assembled contig length to report\r\n",
      "#                                   (def=200)\r\n",
      "#\r\n",
      "#  --long_reads <string>           :fasta file containing error-corrected or circular consensus (CCS) pac bio reads\r\n",
      "#                                   (** note: experimental parameter **, this functionality continues to be under development)\r\n",
      "#\r\n",
      "#  --genome_guided_bam <string>    :genome guided mode, provide path to coordinate-sorted bam file.\r\n",
      "#                                   (see genome-guided param section under --show_full_usage_info)\r\n",
      "#\r\n",
      "#  --jaccard_clip                  :option, set if you have paired reads and\r\n",
      "#                                   you expect high gene density with UTR\r\n",
      "#                                   overlap (use FASTQ input file format\r\n",
      "#                                   for reads).\r\n",
      "#                                   (note: jaccard_clip is an expensive\r\n",
      "#                                   operation, so avoid using it unless\r\n",
      "#                                   necessary due to finding excessive fusion\r\n",
      "#                                   transcripts w/o it.)\r\n",
      "#\r\n",
      "#  --trimmomatic                   :run Trimmomatic to quality trim reads\r\n",
      "#                                        see '--quality_trimming_params' under full usage info for tailored settings.\r\n",
      "#                                  \r\n",
      "#\r\n",
      "#  --no_normalize_reads            :Do *not* run in silico normalization of reads. Defaults to max. read coverage of 50.\r\n",
      "#                                       see '--normalize_max_read_cov' under full usage info for tailored settings.\r\n",
      "#                                       (note, as of Sept 21, 2016, normalization is on by default)\r\n",
      "#     \r\n",
      "#  --no_distributed_trinity_exec   :do not run Trinity phase 2 (assembly of partitioned reads), and stop after generating command list.\r\n",
      "#\r\n",
      "#\r\n",
      "#  --output <string>               :name of directory for output (will be\r\n",
      "#                                   created if it doesn't already exist)\r\n",
      "#                                   default( your current working directory: \"/home/ubuntu/code/sequence-localization/notebooks/trinity_out_dir\" \r\n",
      "#                                    note: must include 'trinity' in the name as a safety precaution! )\r\n",
      "#             \r\n",
      "#  --workdir <string>              :where Trinity phase-2 assembly computation takes place (defaults to --output setting).\r\n",
      "#                                  (can set this to a node-local drive or RAM disk)     \r\n",
      "#  \r\n",
      "#  --full_cleanup                  :only retain the Trinity fasta file, rename as ${output_dir}.Trinity.fasta\r\n",
      "#\r\n",
      "#  --cite                          :show the Trinity literature citation\r\n",
      "#\r\n",
      "#  --verbose                       :provide additional job status info during the run.\r\n",
      "#\r\n",
      "#  --version                       :reports Trinity version (Trinity-v2.5.1) and exits.\r\n",
      "#\r\n",
      "#  --show_full_usage_info          :show the many many more options available for running Trinity (expert usage).\r\n",
      "#\r\n",
      "#\r\n",
      "###############################################################################\r\n",
      "#\r\n",
      "#  *Note, a typical Trinity command might be:\r\n",
      "#\r\n",
      "#        Trinity --seqType fq --max_memory 50G --left reads_1.fq  --right reads_2.fq --CPU 6\r\n",
      "#\r\n",
      "#\r\n",
      "#    and for Genome-guided Trinity:\r\n",
      "#\r\n",
      "#        Trinity --genome_guided_bam rnaseq_alignments.csorted.bam --max_memory 50G\r\n",
      "#                --genome_guided_max_intron 10000 --CPU 6\r\n",
      "#\r\n",
      "#     see: /home/ubuntu/anaconda/opt/trinity-2.5.1/sample_data/test_Trinity_Assembly/\r\n",
      "#          for sample data and 'runMe.sh' for example Trinity execution\r\n",
      "#\r\n",
      "#     For more details, visit: http://trinityrnaseq.github.io\r\n",
      "#\r\n",
      "###############################################################################\r\n",
      "\r\n",
      "\r\n",
      "    \r\n"
     ]
    }
   ],
   "source": [
    "! Trinity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "     ______  ____   ____  ____   ____  ______  __ __\n",
      "    |      ||    \\ |    ||    \\ |    ||      ||  |  |\n",
      "    |      ||  D  ) |  | |  _  | |  | |      ||  |  |\n",
      "    |_|  |_||    /  |  | |  |  | |  | |_|  |_||  ~  |\n",
      "      |  |  |    \\  |  | |  |  | |  |   |  |  |___, |\n",
      "      |  |  |  .  \\ |  | |  |  | |  |   |  |  |     |\n",
      "      |__|  |__|\\_||____||__|__||____|  |__|  |____/\n",
      "\n",
      "\n",
      "Left read files: $VAR1 = [\n",
      "          '/mnt/data/fastq/rnaseq/SRR629563_pass_1.fastq.gz'\n",
      "        ];\n",
      "Right read files: $VAR1 = [\n",
      "          '/mnt/data/fastq/rnaseq/SRR629563_pass_2.fastq.gz'\n",
      "        ];\n",
      "Trinity version: Trinity-v2.5.1\n",
      "** NOTE: Latest version of Trinity is Trinity-v2.6.6, and can be obtained at:\n",
      "\thttps://github.com/trinityrnaseq/trinityrnaseq/releases\n",
      "\n",
      "Paired mode requires bowtie2. Found bowtie2 at: /home/ubuntu/anaconda/bin/bowtie2\n",
      "\n",
      " and bowtie-build at /home/ubuntu/anaconda/bin/bowtie2-build\n",
      "\n",
      "\n",
      "-since butterfly will eventually be run, lets test for proper execution of java\n",
      "Found samtools at: /home/ubuntu/anaconda/bin/samtools\n",
      "\n",
      "#######################################\n",
      "Running Java Tests\n",
      "Wednesday, May 16, 2018: 19:53:04\tCMD: java -Xmx64m -XX:ParallelGCThreads=2  -jar /home/ubuntu/anaconda/opt/trinity-2.5.1/util/support_scripts/ExitTester.jar 0\n",
      "CMD finished (0 seconds)\n",
      "Wednesday, May 16, 2018: 19:53:04\tCMD: java -Xmx64m -XX:ParallelGCThreads=2  -jar /home/ubuntu/anaconda/opt/trinity-2.5.1/util/support_scripts/ExitTester.jar 1\n",
      "-we properly captured the java failure status, as needed.  Looking good.\n",
      "Java tests succeeded.\n",
      "###################################\n",
      "\n",
      "-changing dir to output dir: /mnt/data/trinity-output/fastq_dump_v2/u2os/trinity-output\n",
      "\n",
      "\n",
      "----------------------------------------------------------------------------------\n",
      "-------------- Trinity Phase 1: Clustering of RNA-Seq Reads  ---------------------\n",
      "----------------------------------------------------------------------------------\n",
      "\n",
      "---------------------------------------------------------------\n",
      "------------ In silico Read Normalization ---------------------\n",
      "-- (Removing Excess Reads Beyond 50 Coverage --\n",
      "---------------------------------------------------------------\n",
      "\n",
      "# running normalization on reads: $VAR1 = [\n",
      "          [\n",
      "            '/mnt/data/fastq/rnaseq/SRR629563_pass_1.fastq.gz'\n",
      "          ],\n",
      "          [\n",
      "            '/mnt/data/fastq/rnaseq/SRR629563_pass_2.fastq.gz'\n",
      "          ]\n",
      "        ];\n",
      "\n",
      "\n",
      "-file doesn't yet exist: /mnt/data/trinity-output/fastq_dump_v2/u2os/trinity-output/insilico_read_normalization/left.norm.fq\n",
      "Wednesday, May 16, 2018: 19:53:04\tCMD: /home/ubuntu/anaconda/opt/trinity-2.5.1/util/insilico_read_normalization.pl --seqType fq --JM 16G  --max_cov 50 --CPU 6 --output /mnt/data/trinity-output/fastq_dump_v2/u2os/trinity-output/insilico_read_normalization   --max_pct_stdev 10000  --left /mnt/data/fastq/rnaseq/SRR629563_pass_1.fastq.gz --right /mnt/data/fastq/rnaseq/SRR629563_pass_2.fastq.gz --pairs_together --PARALLEL_STATS  \n",
      "Converting input files. (both directions in parallel)CMD: seqtk-trinity seq -A <(gunzip -c /mnt/data/fastq/rnaseq/SRR629563_pass_1.fastq.gz) >> left.fa\n",
      "CMD: seqtk-trinity seq -A <(gunzip -c /mnt/data/fastq/rnaseq/SRR629563_pass_2.fastq.gz) >> right.fa\n",
      "Error, not recognizing read name formatting: [SRR629563.1.2]\n",
      "\n",
      "If your data come from SRA, be sure to dump the fastq file like so:\n",
      "\n",
      "\tSRA_TOOLKIT/fastq-dump --defline-seq '@$sn[_$rn]/$ri' --split-files file.sra \n",
      "\n",
      "Thread 2 terminated abnormally: Error, cmd: seqtk-trinity seq -A <(gunzip -c /mnt/data/fastq/rnaseq/SRR629563_pass_2.fastq.gz) >> right.fa died with ret 512 at /home/ubuntu/anaconda/opt/trinity-2.5.1/util/insilico_read_normalization.pl line 758.\n",
      "Error, not recognizing read name formatting: [SRR629563.1.1]\n",
      "\n",
      "If your data come from SRA, be sure to dump the fastq file like so:\n",
      "\n",
      "\tSRA_TOOLKIT/fastq-dump --defline-seq '@$sn[_$rn]/$ri' --split-files file.sra \n",
      "\n",
      "Thread 1 terminated abnormally: Error, cmd: seqtk-trinity seq -A <(gunzip -c /mnt/data/fastq/rnaseq/SRR629563_pass_1.fastq.gz) >> left.fa died with ret 512 at /home/ubuntu/anaconda/opt/trinity-2.5.1/util/insilico_read_normalization.pl line 758.\n",
      "Error, conversion thread failed at /home/ubuntu/anaconda/opt/trinity-2.5.1/util/insilico_read_normalization.pl line 329.\n",
      "Error, cmd: /home/ubuntu/anaconda/opt/trinity-2.5.1/util/insilico_read_normalization.pl --seqType fq --JM 16G  --max_cov 50 --CPU 6 --output /mnt/data/trinity-output/fastq_dump_v2/u2os/trinity-output/insilico_read_normalization   --max_pct_stdev 10000  --left /mnt/data/fastq/rnaseq/SRR629563_pass_1.fastq.gz --right /mnt/data/fastq/rnaseq/SRR629563_pass_2.fastq.gz --pairs_together --PARALLEL_STATS   died with ret 7424 at /home/ubuntu/anaconda/bin/Trinity line 2544.\n",
      "\tmain::process_cmd(\"/home/ubuntu/anaconda/opt/trinity-2.5.1/util/insilico_read_no\"...) called at /home/ubuntu/anaconda/bin/Trinity line 3090\n",
      "\tmain::normalize(\"/mnt/data/trinity-output/fastq_dump_v2/u2os/trinity-output/in\"..., 50, ARRAY(0xffc480), ARRAY(0xffc4c8)) called at /home/ubuntu/anaconda/bin/Trinity line 3037\n",
      "\tmain::run_normalization(50, ARRAY(0xffc480), ARRAY(0xffc4c8)) called at /home/ubuntu/anaconda/bin/Trinity line 1297\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# SRR629563 = U-2 OS cell line\n",
    "Trinity --seqType fq --max_memory 16G \\\n",
    "    --left /mnt/data/fastq/rnaseq/SRR629563_pass_1.fastq.gz \\\n",
    "    --right /mnt/data/fastq/rnaseq/SRR629563_pass_2.fastq.gz \\\n",
    "    --output /mnt/data/trinity-output/fastq_dump_v2/u2os/trinity-output \\\n",
    "    --CPU 8 --verbose --workdir /mnt/data/trinity-workdir"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Try properly fastq-dumped data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%bash\n",
    "# SRR629563 = U-2 OS cell line\n",
    "Trinity --seqType fq --max_memory 60G \\\n",
    "    --left /mnt/data/fastq/rnaseq/SRR629563_pass_1.fastq.gz \\\n",
    "    --right /mnt/data/fastq/rnaseq/SRR629563_pass_2.fastq.gz \\\n",
    "    --output /mnt/data/trinity-output/fastq_dump_v2/u2os/trinity-output \\\n",
    "    --CPU 16 --verbose --workdir /mnt/data/trinity-workdir"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
