{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "\n",
    "DATA_FOLDER = os.path.abspath(os.path.join('..', 'data'))\n",
    "FIGURE_FOLDER = os.path.abspath(os.path.join('..', 'figures'))\n",
    "\n",
    "notebook_name = '052_run_cdna_keras_genomics_pyrimidine_purine'\n",
    "\n",
    "data_folder = os.path.join(DATA_FOLDER, notebook_name)\n",
    "figure_folder = os.path.join(FIGURE_FOLDER, notebook_name)\n",
    "\n",
    "! mkdir -p $data_folder\n",
    "! mkdir -p $figure_folder\n",
    "\n",
    "input_folder = os.path.join(DATA_FOLDER, '023_make_train_valid_test_sets')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/src/myhome/code/sequence-localization/data/052_run_cdna_keras_genomics_pyrimidine_purine'"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_folder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets'"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "input_folder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 25M\r\n",
      "drwxr-xr-x  2 root root 4.0K May 26 03:21 \u001b[0m\u001b[01;34m.\u001b[0m/\r\n",
      "drwxrwxr-x 18 2002 2002 4.0K May 29 17:10 \u001b[01;34m..\u001b[0m/\r\n",
      "-rw-r--r--  1 root root  16K May 26 03:21 Homo_sapiens.GRCh38.cdna.all.fa.nuclear_or_cytoplasmic.last500.target.tsv.test\r\n",
      "-rw-r--r--  1 root root  53K May 26 03:21 Homo_sapiens.GRCh38.cdna.all.fa.nuclear_or_cytoplasmic.last500.target.tsv.train\r\n",
      "-rw-r--r--  1 root root 7.6K May 26 03:21 Homo_sapiens.GRCh38.cdna.all.fa.nuclear_or_cytoplasmic.last500.target.tsv.valid\r\n",
      "-rw-r--r--  1 root root 2.9M May 26 03:21 Homo_sapiens.GRCh38.cdna.all.fa.nuclear_or_cytoplasmic.last500.test\r\n",
      "-rw-r--r--  1 root root 9.9M May 26 03:21 Homo_sapiens.GRCh38.cdna.all.fa.nuclear_or_cytoplasmic.last500.train\r\n",
      "-rw-r--r--  1 root root 1.5M May 26 03:21 Homo_sapiens.GRCh38.cdna.all.fa.nuclear_or_cytoplasmic.last500.valid\r\n",
      "-rw-r--r--  1 root root 7.1K May 26 03:21 Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.target.tsv.test\r\n",
      "-rw-r--r--  1 root root  25K May 26 03:21 Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.target.tsv.train\r\n",
      "-rw-r--r--  1 root root 3.6K May 26 03:21 Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.target.tsv.valid\r\n",
      "-rw-r--r--  1 root root 1.3M May 26 03:21 Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.test\r\n",
      "-rw-r--r--  1 root root 4.3M May 26 03:21 Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.train\r\n",
      "-rw-r--r--  1 root root 625K May 26 03:21 Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.valid\r\n",
      "-rw-r--r--  1 root root 7.1K May 26 03:21 Homo_sapiens.GRCh38.pep.all.fa.nuclear_or_cytoplasmic.last150.target.tsv.test\r\n",
      "-rw-r--r--  1 root root  25K May 26 03:21 Homo_sapiens.GRCh38.pep.all.fa.nuclear_or_cytoplasmic.last150.target.tsv.train\r\n",
      "-rw-r--r--  1 root root 3.6K May 26 03:21 Homo_sapiens.GRCh38.pep.all.fa.nuclear_or_cytoplasmic.last150.target.tsv.valid\r\n",
      "-rw-r--r--  1 root root 760K May 26 03:21 Homo_sapiens.GRCh38.pep.all.fa.nuclear_or_cytoplasmic.last150.test\r\n",
      "-rw-r--r--  1 root root 2.6M May 26 03:21 Homo_sapiens.GRCh38.pep.all.fa.nuclear_or_cytoplasmic.last150.train\r\n",
      "-rw-r--r--  1 root root 380K May 26 03:21 Homo_sapiens.GRCh38.pep.all.fa.nuclear_or_cytoplasmic.last150.valid\r\n"
     ]
    }
   ],
   "source": [
    "ls -lha $input_folder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==> /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cdna.all.fa.nuclear_or_cytoplasmic.last500.target.tsv.test <==\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "1\t0\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "\r\n",
      "==> /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cdna.all.fa.nuclear_or_cytoplasmic.last500.target.tsv.train <==\r\n",
      "0\t1\r\n",
      "1\t0\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "1\t0\r\n",
      "0\t1\r\n",
      "\r\n",
      "==> /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cdna.all.fa.nuclear_or_cytoplasmic.last500.target.tsv.valid <==\r\n",
      "1\t0\r\n",
      "1\t0\r\n",
      "1\t0\r\n",
      "1\t0\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "1\t0\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "1\t0\r\n",
      "\r\n",
      "==> /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cdna.all.fa.nuclear_or_cytoplasmic.last500.test <==\r\n",
      ">ENST00000550053.5 cdna chromosome:GRCh38:12:105107715:105169130:1 gene:ENSG00000136051.14 gene_biotype:protein_coding transcript_biotype:nonsense_mediated_decay gene_symbol:WASHC4 description:WASH complex subunit 4 [Source:HGNC Symbol;Acc:HGNC:29174]\r\n",
      "CAATACTAGATAGAGTTCACATGCTGACTCCCTGGATACCTACAGTTGAGAAGAAAATGA\r\n",
      "CAGACAATTATTTTTCCTTTAGGTCAAAATAATGTCACTGCTATCTGGATAAATTATAAG\r\n",
      "TTCATATATTTGAAATAAGATCCTGGAAACGAATTTAAATGGTGTCAAACTGCAGAGCAA\r\n",
      "CAGGGCTTCAGTATATTCACATCTAATAGGTTAATATGTACATTTGGATGATGATTTTTC\r\n",
      "TTGATAACATTAACATTAATTTTATTTTTTCAGACATACTGTAATTTAAAAATTAATGTG\r\n",
      "AGCGTTTGTTTTAAGTTCACAGAAAACTTAATAATTCAAATTCTCAGGAATTAAGGACTC\r\n",
      "ACCATAACATTGTCAGTTCTAATGAAATTTTGTAAAAGATGGCAAGTTTGTTACCTCACT\r\n",
      "TGAGTGGGGTTTTCCTTTTCCCCCAATTCTAAGAGAATATAATGTGTATACTAAAACATT\r\n",
      "AATAAACATAATTTTGAAAA\r\n",
      "\r\n",
      "==> /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cdna.all.fa.nuclear_or_cytoplasmic.last500.train <==\r\n",
      ">ENST00000419349.2 cdna chromosome:GRCh38:3:49357178:49358312:-1 gene:ENSG00000233276.4 gene_biotype:polymorphic_pseudogene transcript_biotype:protein_coding gene_symbol:GPX1 description:glutathione peroxidase 1 [Source:HGNC Symbol;Acc:HGNC:4553]\r\n",
      "ACTTCATGCTCTTCGAGAAGTGCGAGGTGAACGGTGCGGGGGCGCACCCTCTCTTCGCCT\r\n",
      "TCCTGCGGGAGGCCCTGCCAGCTCCCAGCGACGACGCCACCGCGCTTATGACCGACCCCA\r\n",
      "AGCTCATCACCTGGTCTCCGGTGTGTCGCAACGATGTTGCCTGGAACTTTGAGAAGTTCC\r\n",
      "TGGTGGGCCCTGACGGTGTGCCCCTACGCAGGTACAGCCGCCGCTTCCAGACCATTGACA\r\n",
      "TCGAGCCTGACATCGAAGCCCTGCTGTCTCAAGGGCCCAGCTGTGCCTAGGGCGCCCCTC\r\n",
      "CTACCCCGGCTGCTTGGCAGTTGCAGTGCTGCTGTCTCGGGGGGGTTTTCATCTATGAGG\r\n",
      "GTGTTTCCTCTAAACCTACGAGGGAGGAACACCTGATCTTACAGAAAATACCACCTCGAG\r\n",
      "ATGGGTGCTGGTCCTGTTGATCCCAGTCTCTGCCAGACCAAGGCGAGTTTCCCCACTAAT\r\n",
      "AAAGTGCCGGGTGTCAGCAG\r\n",
      "\r\n",
      "==> /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cdna.all.fa.nuclear_or_cytoplasmic.last500.valid <==\r\n",
      ">ENST00000419783.2 cdna chromosome:GRCh38:3:49357171:49358600:-1 gene:ENSG00000233276.4 gene_biotype:polymorphic_pseudogene transcript_biotype:protein_coding gene_symbol:GPX1 description:glutathione peroxidase 1 [Source:HGNC Symbol;Acc:HGNC:4553]\r\n",
      "GCTCTTCGAGAAGTGCGAGGTGAACGGTGCGGGGGCGCACCCTCTCTTCGCCTTCCTGCG\r\n",
      "GGAGGCCCTGCCAGCTCCCAGCGACGACGCCACCGCGCTTATGACCGACCCCAAGCTCAT\r\n",
      "CACCTGGTCTCCGGTGTGTCGCAACGATGTTGCCTGGAACTTTGAGAAGTTCCTGGTGGG\r\n",
      "CCCTGACGGTGTGCCCCTACGCAGGTACAGCCGCCGCTTCCAGACCATTGACATCGAGCC\r\n",
      "TGACATCGAAGCCCTGCTGTCTCAAGGGCCCAGCTGTGCCTAGGGCGCCCCTCCTACCCC\r\n",
      "GGCTGCTTGGCAGTTGCAGTGCTGCTGTCTCGGGGGGGTTTTCATCTATGAGGGTGTTTC\r\n",
      "CTCTAAACCTACGAGGGAGGAACACCTGATCTTACAGAAAATACCACCTCGAGATGGGTG\r\n",
      "CTGGTCCTGTTGATCCCAGTCTCTGCCAGACCAAGGCGAGTTTCCCCACTAATAAAGTGC\r\n",
      "CGGGTGTCAGCAGAACTGTG\r\n",
      "\r\n",
      "==> /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.target.tsv.test <==\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "\r\n",
      "==> /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.target.tsv.train <==\r\n",
      "0\t1\r\n",
      "1\t0\r\n",
      "0\t1\r\n",
      "1\t0\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "1\t0\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "\r\n",
      "==> /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.target.tsv.valid <==\r\n",
      "1\t0\r\n",
      "0\t1\r\n",
      "1\t0\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "1\t0\r\n",
      "1\t0\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "\r\n",
      "==> /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.test <==\r\n",
      ">ENST00000367603.7 cds chromosome:GRCh38:6:142751467:142945201:-1 gene:ENSG00000010818.9 gene_biotype:protein_coding transcript_biotype:protein_coding gene_symbol:HIVEP2 description:human immunodeficiency virus type I enhancer binding protein 2 [Source:HGNC Symbol;Acc:HGNC:4921]\r\n",
      "ACTCCCTCCTCTCCTCGGCTGTTGATGAAACAGAGCACTTCGGAAGACAGCCTAAACGCA\r\n",
      "ACAGAGCGGGAACAGGAGGAAAATATACAGACTTGTACAAAAGCCATTGCCTCTCTCCGG\r\n",
      "ATTGCCACGGAAGAGGCAGCTCTGCTCGGGCCAGATCAGCCAGCGCGGGTGCAGGAGCCC\r\n",
      "CACCAGAACCCCCTGGGAAGTGCACATGTTAGCATTAGACACTTTAGTAGACCTGAGCCA\r\n",
      "GGTCAGCCCTGTACCTCAGCCACCCACCCTGACTTGCATGATGGTGAAAAGGACAATTTT\r\n",
      "GGTACATCACAGACTCCATTAGCTCACTCCACGTTTTACAGCAAGAGTTGTGTGGATGAC\r\n",
      "AAGCAGTTGGACTTTCACAGCAGCAAGGAATTATCTTCAAGCACAGAGGAAAGCAAAGAT\r\n",
      "CCTTCATCAGAAAAGAGTCAGCTACATTGA\r\n",
      ">ENST00000367604.6 cds chromosome:GRCh38:6:142751478:142946365:-1 gene:ENSG00000010818.9 gene_biotype:protein_coding transcript_biotype:protein_coding gene_symbol:HIVEP2 description:human immunodeficiency virus type I enhancer binding protein 2 [Source:HGNC Symbol;Acc:HGNC:4921]\r\n",
      "\r\n",
      "==> /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.train <==\r\n",
      ">ENST00000419783.2 cds chromosome:GRCh38:3:49357171:49358600:-1 gene:ENSG00000233276.4 gene_biotype:polymorphic_pseudogene transcript_biotype:protein_coding gene_symbol:GPX1 description:glutathione peroxidase 1 [Source:HGNC Symbol;Acc:HGNC:4553]\r\n",
      "GACTACACCCAGATGAACGAGCTGCAGCGGCGCCTCGGACCCCGGGGCCTGGTGGTGCTC\r\n",
      "GGCTTCCCGTGCAACCAGTTTGGGCATCAGGAGAACGCCAAGAACGAAGAGATTCTGAAT\r\n",
      "TCCCTCAAGTACGTCCGGCCTGGTGGTGGGTTCGAGCCCAACTTCATGCTCTTCGAGAAG\r\n",
      "TGCGAGGTGAACGGTGCGGGGGCGCACCCTCTCTTCGCCTTCCTGCGGGAGGCCCTGCCA\r\n",
      "GCTCCCAGCGACGACGCCACCGCGCTTATGACCGACCCCAAGCTCATCACCTGGTCTCCG\r\n",
      "GTGTGTCGCAACGATGTTGCCTGGAACTTTGAGAAGTTCCTGGTGGGCCCTGACGGTGTG\r\n",
      "CCCCTACGCAGGTACAGCCGCCGCTTCCAGACCATTGACATCGAGCCTGACATCGAAGCC\r\n",
      "CTGCTGTCTCAAGGGCCCAGCTGTGCCTAG\r\n",
      ">ENST00000643797.1 cds chromosome:GRCh38:3:49357201:49358325:-1 gene:ENSG00000233276.4 gene_biotype:polymorphic_pseudogene transcript_biotype:protein_coding gene_symbol:GPX1 description:glutathione peroxidase 1 [Source:HGNC Symbol;Acc:HGNC:4553]\r\n",
      "\r\n",
      "==> /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.valid <==\r\n",
      ">ENST00000266269.9 cds chromosome:GRCh38:22:31325804:31346232:-1 gene:ENSG00000100105.17 gene_biotype:protein_coding transcript_biotype:protein_coding gene_symbol:PATZ1 description:POZ/BTB and AT hook containing zinc finger 1 [Source:HGNC Symbol;Acc:HGNC:13071]\r\n",
      "TTCCACTGCGCCAGGACCTATGGCAACAAAGAAGGCCAGAAATGCTCACATCAGGATCCG\r\n",
      "ATTGAGAGCTCTGACTCCTATGGTGACCTCTCAGATGCCAGCGACCTGAAGACGCCAGAG\r\n",
      "AAGCAGAGTGCCAATGGCTCTTTCTCCTGCGACATGGCAGTCCCCAAAAACAAAATGGAG\r\n",
      "TCTGATGGGGAGAAGAAGTACCCATGCCCTGAATGTGGGAGCTTCTTCCGCTCTAAGTCC\r\n",
      "TACTTGAACAAACACATCCAGAAGGTGCATGTCCGGGCTCTCGGGGGCCCCCTGGGGGAC\r\n",
      "CTGGGCCCTGCCCTTGGCTCACCTTTCTCTCCTCAGCAGAACATGTCTCTCCTCGAGTCC\r\n",
      "TTTGGGTTTCAGATTGTTCAGTCGGCATTTGCGTCATCTTTAGTAGATCCTGAGGTTGAC\r\n",
      "CAGCAGCCCATGGGGCCTGAAGGGAAATGA\r\n",
      ">ENST00000426431.2 cds chromosome:GRCh38:12:53380639:53416446:1 gene:ENSG00000185591.9 gene_biotype:protein_coding transcript_biotype:protein_coding gene_symbol:SP1 description:Sp1 transcription factor [Source:HGNC Symbol;Acc:HGNC:11205]\r\n",
      "\r\n",
      "==> /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.pep.all.fa.nuclear_or_cytoplasmic.last150.target.tsv.test <==\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "1\t0\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "\r\n",
      "==> /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.pep.all.fa.nuclear_or_cytoplasmic.last150.target.tsv.train <==\r\n",
      "1\t0\r\n",
      "1\t0\r\n",
      "1\t0\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "1\t0\r\n",
      "1\t0\r\n",
      "1\t0\r\n",
      "1\t0\r\n",
      "\r\n",
      "==> /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.pep.all.fa.nuclear_or_cytoplasmic.last150.target.tsv.valid <==\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "0\t1\r\n",
      "1\t0\r\n",
      "0\t1\r\n",
      "1\t0\r\n",
      "1\t0\r\n",
      "0\t1\r\n",
      "\r\n",
      "==> /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.pep.all.fa.nuclear_or_cytoplasmic.last150.test <==\r\n",
      ">ENSP00000356575.2 pep chromosome:GRCh38:6:142751467:142945201:-1 gene:ENSG00000010818.9 transcript:ENST00000367603.7 gene_biotype:protein_coding transcript_biotype:protein_coding gene_symbol:HIVEP2 description:human immunodeficiency virus type I enhancer binding protein 2 [Source:HGNC Symbol;Acc:HGNC:4921]\r\n",
      "STPSSPRLLMKQSTSEDSLNATEREQEENIQTCTKAIASLRIATEEAALLGPDQPARVQE\r\n",
      "PHQNPLGSAHVSIRHFSRPEPGQPCTSATHPDLHDGEKDNFGTSQTPLAHSTFYSKSCVD\r\n",
      "DKQLDFHSSKELSSSTEESKDPSSEKSQLH\r\n",
      ">ENSP00000215919.3 pep chromosome:GRCh38:22:31340677:31345776:-1 gene:ENSG00000100105.17 transcript:ENST00000215919.3 gene_biotype:protein_coding transcript_biotype:protein_coding gene_symbol:PATZ1 description:POZ/BTB and AT hook containing zinc finger 1 [Source:HGNC Symbol;Acc:HGNC:13071]\r\n",
      "CGLRFKRKDRMSYHVRSHDGSVGKPYICQSCGKGFSRPDHLNGHIKQVHTSERPHKCQVW\r\n",
      "VGSSSGLPPLEPLPSDLPSWDFAQPALWRSSHSVPDTAFSLSLKKSFPALENLGPAHSSN\r\n",
      "TLFCPAPPGYLRQGWTTPEGSRAFTQWPVG\r\n",
      ">ENSP00000388784.1 pep chromosome:GRCh38:19:11887818:11904902:1 gene:ENSG00000198429.9 transcript:ENST00000445911.5 gene_biotype:protein_coding transcript_biotype:protein_coding gene_symbol:ZNF69 description:zinc finger protein 69 [Source:HGNC Symbol;Acc:HGNC:13138]\r\n",
      "SHRRCREDPGTSESQEMEEWALLDISQRKLYKEVMLETFRNLTSVGKSWKDQNIEYEYQN\r\n",
      "\r\n",
      "==> /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.pep.all.fa.nuclear_or_cytoplasmic.last150.train <==\r\n",
      ">ENSP00000495108.1 pep chromosome:GRCh38:3:49357201:49358325:-1 gene:ENSG00000233276.4 transcript:ENST00000643797.1 gene_biotype:polymorphic_pseudogene transcript_biotype:protein_coding gene_symbol:GPX1 description:glutathione peroxidase 1 [Source:HGNC Symbol;Acc:HGNC:4553]\r\n",
      "AAAAAAQSVYAFSARPLAGGEPVSLGSLRGKENAKNEEILNSLKYVRPGGGFEPNFMLFE\r\n",
      "KCEVNGAGAHPLFAFLREALPAPSDDATALMTDPKLITWSPVCRNDVAWNFEKFLVGPDG\r\n",
      "VPLRRYSRRFQTIDIEPDIEALLSQGPSCA\r\n",
      ">ENSP00000495001.1 pep chromosome:GRCh38:3:49357477:49358325:-1 gene:ENSG00000233276.4 transcript:ENST00000646881.1 gene_biotype:polymorphic_pseudogene transcript_biotype:polymorphic_pseudogene gene_symbol:GPX1 description:glutathione peroxidase 1 [Source:HGNC Symbol;Acc:HGNC:4553]\r\n",
      "ARPLAGGEPVSLGSLRGKVLLIENVASL*GTTVRDYTQMNELQRRLGPRGLVVLGFPCNQ\r\n",
      "FGHQLPFSPVGERQERRDSEFPQVRPAWWWVRAQLHALREVRGERCGGAPSLRLPAGGPA\r\n",
      "SSQRRRHRAYDRPQAHHLVSGVSQRCCLEL\r\n",
      ">ENSP00000478837.1 pep chromosome:GRCh38:3:49357180:49358358:-1 gene:ENSG00000233276.4 transcript:ENST00000620890.1 gene_biotype:polymorphic_pseudogene transcript_biotype:protein_coding gene_symbol:GPX1 description:glutathione peroxidase 1 [Source:HGNC Symbol;Acc:HGNC:4553]\r\n",
      "RDYTQMNELQRRLGPRGLVVLGFPCNQFGHQENAKNEEILNSLKYVRPGGGFEPNFMLFE\r\n",
      "\r\n",
      "==> /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.pep.all.fa.nuclear_or_cytoplasmic.last150.valid <==\r\n",
      ">ENSP00000407375.1 pep chromosome:GRCh38:3:49357171:49358600:-1 gene:ENSG00000233276.4 transcript:ENST00000419783.2 gene_biotype:polymorphic_pseudogene transcript_biotype:protein_coding gene_symbol:GPX1 description:glutathione peroxidase 1 [Source:HGNC Symbol;Acc:HGNC:4553]\r\n",
      "RDYTQMNELQRRLGPRGLVVLGFPCNQFGHQENAKNEEILNSLKYVRPGGGFEPNFMLFE\r\n",
      "KCEVNGAGAHPLFAFLREALPAPSDDATALMTDPKLITWSPVCRNDVAWNFEKFLVGPDG\r\n",
      "VPLRRYSRRFQTIDIEPDIEALLSQGPSCA\r\n",
      ">ENSP00000295321.4 pep chromosome:GRCh38:2:127480806:127526468:-1 gene:ENSG00000163166.14 transcript:ENST00000295321.8 gene_biotype:protein_coding transcript_biotype:protein_coding gene_symbol:IWS1 description:IWS1, SUPT6H interacting protein [Source:HGNC Symbol;Acc:HGNC:25467]\r\n",
      "ESRSNKDMAGKLINEWSRPIFGLTSNYKGMTREEREQRDLEQMPQRRRMNSTGGQTPRRD\r\n",
      "LEKVLTGEEKALRPGDPGFCARARVPMPSNKDYVVRPKWNVEMESSRFQATSKKGISRLD\r\n",
      "KQMRKFTDIRKKSRSAHAVKISIEGNKMPL\r\n",
      ">ENSP00000377679.4 pep chromosome:GRCh38:17:40062193:40093613:1 gene:ENSG00000126351.12 transcript:ENST00000394121.8 gene_biotype:protein_coding transcript_biotype:protein_coding gene_symbol:THRA description:thyroid hormone receptor, alpha [Source:HGNC Symbol;Acc:HGNC:11796]\r\n",
      "SQEAYLLAFEHYVNHRKHNIPHFWPKLLMKEREVQSSILYKGAAAEGRPGGSLGVHPEGQ\r\n"
     ]
    }
   ],
   "source": [
    "! head $input_folder/*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "# from tensorflow.python.client import device_lib ; print(device_lib.list_local_devices())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "A\t1\t1\r\n",
      "C\t1\t0\r\n",
      "G\t0\t0\r\n",
      "T\t0\t1\r\n"
     ]
    }
   ],
   "source": [
    "cat /src/myhome/code/Keras-genomics/twobit.mapper.tsv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Making train dataset ...\n",
      "\\tCommand: python /src/myhome/code/Keras-genomics/embedH5.py         --mapperfile /src/myhome/code/Keras-genomics/twobit.mapper.tsv         /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.train         /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.target.tsv.train         /src/myhome/code/sequence-localization/data/052_run_cdna_keras_genomics_pyrimidine_purine/cdna_expt01/train.h5\n",
      "args: Namespace(batch=5000, dataname='data', infile='/src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.train', infile2='', isseq='Y', labelfile='/src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.target.tsv.train', labelname='label', maniprefix='/data', mapperfile='/src/myhome/code/Keras-genomics/twobit.mapper.tsv', outfile='/src/myhome/code/sequence-localization/data/052_run_cdna_keras_genomics_pyrimidine_purine/cdna_expt01/train.h5')\n",
      "True\n",
      "batch_array.shape - before slicing (10000,)\n",
      "batch_array.shape - after slicing (6311,)\n",
      "target.shape (6311, 2)\n",
      "seqs.shape (6311, 2)\n",
      "data shape:  (5000, 2, 1, 450)\n",
      "data shape:  (1311, 2, 1, 450)\n",
      "Making valid dataset ...\n",
      "\\tCommand: python /src/myhome/code/Keras-genomics/embedH5.py         --mapperfile /src/myhome/code/Keras-genomics/twobit.mapper.tsv         /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.valid         /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.target.tsv.valid         /src/myhome/code/sequence-localization/data/052_run_cdna_keras_genomics_pyrimidine_purine/cdna_expt01/valid.h5\n",
      "args: Namespace(batch=5000, dataname='data', infile='/src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.valid', infile2='', isseq='Y', labelfile='/src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.target.tsv.valid', labelname='label', maniprefix='/data', mapperfile='/src/myhome/code/Keras-genomics/twobit.mapper.tsv', outfile='/src/myhome/code/sequence-localization/data/052_run_cdna_keras_genomics_pyrimidine_purine/cdna_expt01/valid.h5')\n",
      "True\n",
      "batch_array.shape - before slicing (5000,)\n",
      "batch_array.shape - after slicing (902,)\n",
      "target.shape (902, 2)\n",
      "seqs.shape (902, 2)\n",
      "data shape:  (902, 2, 1, 450)\n",
      "Making test dataset ...\n",
      "\\tCommand: python /src/myhome/code/Keras-genomics/embedH5.py         --mapperfile /src/myhome/code/Keras-genomics/twobit.mapper.tsv         /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.test         /src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.target.tsv.test         /src/myhome/code/sequence-localization/data/052_run_cdna_keras_genomics_pyrimidine_purine/cdna_expt01/test.h5\n",
      "args: Namespace(batch=5000, dataname='data', infile='/src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.test', infile2='', isseq='Y', labelfile='/src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets/Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450.target.tsv.test', labelname='label', maniprefix='/data', mapperfile='/src/myhome/code/Keras-genomics/twobit.mapper.tsv', outfile='/src/myhome/code/sequence-localization/data/052_run_cdna_keras_genomics_pyrimidine_purine/cdna_expt01/test.h5')\n",
      "True\n",
      "batch_array.shape - before slicing (5000,)\n",
      "batch_array.shape - after slicing (1803,)\n",
      "target.shape (1803, 2)\n",
      "seqs.shape (1803, 2)\n",
      "data shape:  (1803, 2, 1, 450)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.\n",
      "  from ._conv import register_converters as _register_converters\n",
      "/opt/conda/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.\n",
      "  from ._conv import register_converters as _register_converters\n",
      "/opt/conda/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.\n",
      "  from ._conv import register_converters as _register_converters\n"
     ]
    }
   ],
   "source": [
    "%%bash -s \"input_folder='$input_folder'; echo input_folder: $input_folder\"\n",
    "\n",
    "data_folder=/src/myhome/code/sequence-localization/data/052_run_cdna_keras_genomics_pyrimidine_purine\n",
    "input_folder=/src/myhome/code/sequence-localization/data/023_make_train_valid_test_sets\n",
    "EXPT_DIR=$data_folder/cdna_expt01\n",
    "\n",
    "PREFIX='Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.last450'\n",
    "\n",
    "for dtype in 'train' 'valid' 'test'\n",
    "do\n",
    "    echo Making $dtype dataset ...\n",
    "    echo \"\\tCommand: python /src/myhome/code/Keras-genomics/embedH5.py \\\n",
    "        --mapperfile /src/myhome/code/Keras-genomics/twobit.mapper.tsv \\\n",
    "        $input_folder/$PREFIX.$dtype \\\n",
    "        $input_folder/$PREFIX.target.tsv.$dtype \\\n",
    "        $EXPT_DIR/$dtype.h5\"\n",
    "    python /src/myhome/code/Keras-genomics/embedH5.py \\\n",
    "        --mapperfile /src/myhome/code/Keras-genomics/twobit.mapper.tsv\\\n",
    "        $input_folder/$PREFIX.$dtype \\\n",
    "        $input_folder/$PREFIX.target.tsv.$dtype \\\n",
    "        $EXPT_DIR/$dtype.h5\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# %run -d -b 40 /src/myhome/code/Keras-genomics/embedH5.py /src/myhome/code/sequence-localization/data/022_make_train_valid_test_sets/Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.test         /src/myhome/code/sequence-localization/data/022_make_train_valid_test_sets/Homo_sapiens.GRCh38.cds.all.fa.nuclear_or_cytoplasmic.target.tsv.test         /src/myhome/code/sequence-localization/data/040_run_cdna_keras_genomics/cdna_expt01/test.h5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: future in /opt/conda/lib/python3.6/site-packages (0.16.0)\n",
      "\u001b[31mmkl-random 1.0.1 requires cython, which is not installed.\u001b[0m\n",
      "\u001b[31mmkl-fft 1.0.0 requires cython, which is not installed.\u001b[0m\n",
      "\u001b[31mhyperas 0.4 requires jupyter, which is not installed.\u001b[0m\n",
      "\u001b[31mdistributed 1.21.8 requires msgpack, which is not installed.\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "! pip install future"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.\n",
      "  from ._conv import register_converters as _register_converters\n",
      "Using TensorFlow backend.\n",
      "Reading /src/myhome/code/sequence-localization/data/050_run_cdna_keras_genomics/cdna_expt01/train.h5.batch1 ...\n",
      "Reading /src/myhome/code/sequence-localization/data/050_run_cdna_keras_genomics/cdna_expt01/train.h5.batch2 ...\n",
      "Reading /src/myhome/code/sequence-localization/data/050_run_cdna_keras_genomics/cdna_expt01/valid.h5.batch1 ...\n",
      "\n",
      "*** 9 configurations x 2.2 iterations each\n",
      "\n",
      "1 | Tue May 29 16:57:46 2018 | lowest loss so far: inf (run -1)\n",
      "\n",
      "iterations: 2.2222222222222223\n",
      "{'DELTA': 1e-08, 'DROPOUT': 0.5, 'MOMENT': 0.9}\n",
      "\n",
      "Train on 6311 samples, validate on 902 samples\n",
      "Epoch 1/2\n",
      "2018-05-29 16:57:46.631832: I tensorflow/core/platform/cpu_feature_guard.cc:140] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA\n",
      "2018-05-29 16:57:46.982284: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1356] Found device 0 with properties: \n",
      "name: TITAN Xp major: 6 minor: 1 memoryClockRate(GHz): 1.582\n",
      "pciBusID: 0000:02:00.0\n",
      "totalMemory: 11.90GiB freeMemory: 11.74GiB\n",
      "2018-05-29 16:57:47.324155: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1356] Found device 1 with properties: \n",
      "name: TITAN Xp major: 6 minor: 1 memoryClockRate(GHz): 1.582\n",
      "pciBusID: 0000:03:00.0\n",
      "totalMemory: 11.90GiB freeMemory: 11.74GiB\n",
      "2018-05-29 16:57:47.654585: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1356] Found device 2 with properties: \n",
      "name: TITAN Xp major: 6 minor: 1 memoryClockRate(GHz): 1.582\n",
      "pciBusID: 0000:83:00.0\n",
      "totalMemory: 11.90GiB freeMemory: 11.74GiB\n",
      "2018-05-29 16:57:48.008758: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1356] Found device 3 with properties: \n",
      "name: TITAN Xp major: 6 minor: 1 memoryClockRate(GHz): 1.582\n",
      "pciBusID: 0000:84:00.0\n",
      "totalMemory: 11.90GiB freeMemory: 11.74GiB\n",
      "2018-05-29 16:57:48.011176: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1435] Adding visible gpu devices: 0, 1, 2, 3\n",
      "2018-05-29 16:57:49.562615: I tensorflow/core/common_runtime/gpu/gpu_device.cc:923] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2018-05-29 16:57:49.562704: I tensorflow/core/common_runtime/gpu/gpu_device.cc:929]      0 1 2 3 \n",
      "2018-05-29 16:57:49.562720: I tensorflow/core/common_runtime/gpu/gpu_device.cc:942] 0:   N Y N N \n",
      "2018-05-29 16:57:49.562735: I tensorflow/core/common_runtime/gpu/gpu_device.cc:942] 1:   Y N N N \n",
      "2018-05-29 16:57:49.562747: I tensorflow/core/common_runtime/gpu/gpu_device.cc:942] 2:   N N N Y \n",
      "2018-05-29 16:57:49.562759: I tensorflow/core/common_runtime/gpu/gpu_device.cc:942] 3:   N N Y N \n",
      "2018-05-29 16:57:49.564365: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1053] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 11364 MB memory) -> physical GPU (device: 0, name: TITAN Xp, pci bus id: 0000:02:00.0, compute capability: 6.1)\n",
      "2018-05-29 16:57:49.702079: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1053] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:1 with 11364 MB memory) -> physical GPU (device: 1, name: TITAN Xp, pci bus id: 0000:03:00.0, compute capability: 6.1)\n",
      "2018-05-29 16:57:49.856570: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1053] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:2 with 11364 MB memory) -> physical GPU (device: 2, name: TITAN Xp, pci bus id: 0000:83:00.0, compute capability: 6.1)\n",
      "2018-05-29 16:57:50.004298: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1053] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:3 with 11364 MB memory) -> physical GPU (device: 3, name: TITAN Xp, pci bus id: 0000:84:00.0, compute capability: 6.1)\n",
      "6311/6311 [==============================] - 6s 907us/step - loss: 0.6741 - acc: 0.6023 - val_loss: 0.6724 - val_acc: 0.6231\n",
      "Epoch 2/2\n",
      "6311/6311 [==============================] - 0s 78us/step - loss: 0.6637 - acc: 0.6196 - val_loss: 0.6667 - val_acc: 0.6231\n",
      "902/902 [==============================] - 0s 89us/step\n",
      "\n",
      "7 seconds.\n",
      "\n",
      "2 | Tue May 29 16:57:52 2018 | lowest loss so far: 0.6667 (run 1)\n",
      "\n",
      "iterations: 2.2222222222222223\n",
      "{'DELTA': 1e-06, 'DROPOUT': 0.75, 'MOMENT': 0.999}\n",
      "\n",
      "Train on 6311 samples, validate on 902 samples\n",
      "Epoch 1/2\n",
      "6311/6311 [==============================] - 1s 157us/step - loss: 0.7355 - acc: 0.6072 - val_loss: 0.6647 - val_acc: 0.6231\n",
      "Epoch 2/2\n",
      "6311/6311 [==============================] - 0s 76us/step - loss: 0.6659 - acc: 0.6175 - val_loss: 0.6625 - val_acc: 0.6231\n",
      "902/902 [==============================] - 0s 72us/step\n",
      "\n",
      "2 seconds.\n",
      "\n",
      "3 | Tue May 29 16:57:55 2018 | lowest loss so far: 0.6625 (run 2)\n",
      "\n",
      "iterations: 2.2222222222222223\n",
      "{'DELTA': 1e-06, 'DROPOUT': 0.75, 'MOMENT': 0.999}\n",
      "\n",
      "Train on 6311 samples, validate on 902 samples\n",
      "Epoch 1/2\n",
      "6311/6311 [==============================] - 1s 163us/step - loss: 0.6839 - acc: 0.6135 - val_loss: 0.6629 - val_acc: 0.6231\n",
      "Epoch 2/2\n",
      "6311/6311 [==============================] - 0s 78us/step - loss: 0.6657 - acc: 0.6175 - val_loss: 0.6629 - val_acc: 0.6231\n",
      "902/902 [==============================] - 0s 73us/step\n",
      "\n",
      "2 seconds.\n",
      "\n",
      "4 | Tue May 29 16:57:57 2018 | lowest loss so far: 0.6625 (run 2)\n",
      "\n",
      "iterations: 2.2222222222222223\n",
      "{'DELTA': 0.0001, 'DROPOUT': 0.5, 'MOMENT': 0.9}\n",
      "\n",
      "Train on 6311 samples, validate on 902 samples\n",
      "Epoch 1/2\n",
      "6311/6311 [==============================] - 1s 163us/step - loss: 0.6814 - acc: 0.6161 - val_loss: 0.6626 - val_acc: 0.6231\n",
      "Epoch 2/2\n",
      "6311/6311 [==============================] - 0s 77us/step - loss: 0.6657 - acc: 0.6175 - val_loss: 0.6633 - val_acc: 0.6231\n",
      "902/902 [==============================] - 0s 74us/step\n",
      "\n",
      "2 seconds.\n",
      "\n",
      "5 | Tue May 29 16:57:59 2018 | lowest loss so far: 0.6625 (run 2)\n",
      "\n",
      "iterations: 2.2222222222222223\n",
      "{'DELTA': 1e-06, 'DROPOUT': 0.1, 'MOMENT': 0.9}\n",
      "\n",
      "Train on 6311 samples, validate on 902 samples\n",
      "Epoch 1/2\n",
      "6311/6311 [==============================] - 1s 174us/step - loss: 0.6730 - acc: 0.6093 - val_loss: 0.6678 - val_acc: 0.6231\n",
      "Epoch 2/2\n",
      "6311/6311 [==============================] - 0s 73us/step - loss: 0.6646 - acc: 0.6175 - val_loss: 0.6687 - val_acc: 0.6231\n",
      "902/902 [==============================] - 0s 74us/step\n",
      "\n",
      "2 seconds.\n",
      "\n",
      "6 | Tue May 29 16:58:01 2018 | lowest loss so far: 0.6625 (run 2)\n",
      "\n",
      "iterations: 2.2222222222222223\n",
      "{'DELTA': 1e-08, 'DROPOUT': 0.5, 'MOMENT': 0.9}\n",
      "\n",
      "Train on 6311 samples, validate on 902 samples\n",
      "Epoch 1/2\n",
      "6311/6311 [==============================] - 1s 194us/step - loss: 0.6735 - acc: 0.6077 - val_loss: 0.6634 - val_acc: 0.6231\n",
      "Epoch 2/2\n",
      "6311/6311 [==============================] - 0s 78us/step - loss: 0.6657 - acc: 0.6167 - val_loss: 0.6682 - val_acc: 0.6231\n",
      "902/902 [==============================] - 0s 78us/step\n",
      "\n",
      "3 seconds.\n",
      "\n",
      "7 | Tue May 29 16:58:04 2018 | lowest loss so far: 0.6625 (run 2)\n",
      "\n",
      "iterations: 2.2222222222222223\n",
      "{'DELTA': 1e-08, 'DROPOUT': 0.75, 'MOMENT': 0.999}\n",
      "\n",
      "Train on 6311 samples, validate on 902 samples\n",
      "Epoch 1/2\n",
      "6311/6311 [==============================] - 1s 199us/step - loss: 0.6783 - acc: 0.6140 - val_loss: 0.6754 - val_acc: 0.6231\n",
      "Epoch 2/2\n",
      "6311/6311 [==============================] - 0s 78us/step - loss: 0.6699 - acc: 0.6181 - val_loss: 0.6641 - val_acc: 0.6231\n",
      "902/902 [==============================] - 0s 81us/step\n",
      "\n",
      "2 seconds.\n",
      "\n",
      "8 | Tue May 29 16:58:06 2018 | lowest loss so far: 0.6625 (run 2)\n",
      "\n",
      "iterations: 2.2222222222222223\n",
      "{'DELTA': 0.0001, 'DROPOUT': 0.5, 'MOMENT': 0.999}\n",
      "\n",
      "Train on 6311 samples, validate on 902 samples\n",
      "Epoch 1/2\n",
      "6311/6311 [==============================] - 1s 199us/step - loss: 0.6704 - acc: 0.6121 - val_loss: 0.6657 - val_acc: 0.6231\n",
      "Epoch 2/2\n",
      "6311/6311 [==============================] - 1s 80us/step - loss: 0.6669 - acc: 0.6175 - val_loss: 0.6992 - val_acc: 0.3769\n",
      "902/902 [==============================] - 0s 76us/step\n",
      "\n",
      "3 seconds.\n",
      "\n",
      "9 | Tue May 29 16:58:09 2018 | lowest loss so far: 0.6625 (run 2)\n",
      "\n",
      "iterations: 2.2222222222222223\n",
      "{'DELTA': 0.0001, 'DROPOUT': 0.75, 'MOMENT': 0.999}\n",
      "\n",
      "Train on 6311 samples, validate on 902 samples\n",
      "Epoch 1/2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6311/6311 [==============================] - 1s 209us/step - loss: 0.7367 - acc: 0.6086 - val_loss: 0.6674 - val_acc: 0.6231\n",
      "Epoch 2/2\n",
      "6311/6311 [==============================] - 0s 79us/step - loss: 0.6669 - acc: 0.6175 - val_loss: 0.6819 - val_acc: 0.6231\n",
      "902/902 [==============================] - 0s 74us/step\n",
      "\n",
      "3 seconds.\n",
      "\n",
      "*** 3.0 configurations x 6.7 iterations each\n",
      "\n",
      "10 | Tue May 29 16:58:12 2018 | lowest loss so far: 0.6625 (run 2)\n",
      "\n",
      "iterations: 6.666666666666667\n",
      "{'DELTA': 1e-06, 'DROPOUT': 0.75, 'MOMENT': 0.999}\n",
      "\n",
      "Train on 6311 samples, validate on 902 samples\n",
      "Epoch 1/7\n",
      "6311/6311 [==============================] - 1s 209us/step - loss: 0.6908 - acc: 0.6110 - val_loss: 0.6626 - val_acc: 0.6231\n",
      "Epoch 2/7\n",
      "6311/6311 [==============================] - 0s 79us/step - loss: 0.6657 - acc: 0.6175 - val_loss: 0.6632 - val_acc: 0.6231\n",
      "Epoch 3/7\n",
      "6311/6311 [==============================] - 1s 84us/step - loss: 0.6655 - acc: 0.6175 - val_loss: 0.6626 - val_acc: 0.6231\n",
      "Epoch 4/7\n",
      "6311/6311 [==============================] - 1s 81us/step - loss: 0.6655 - acc: 0.6175 - val_loss: 0.6636 - val_acc: 0.6231\n",
      "902/902 [==============================] - 0s 78us/step\n",
      "\n",
      "4 seconds.\n",
      "\n",
      "11 | Tue May 29 16:58:15 2018 | lowest loss so far: 0.6625 (run 2)\n",
      "\n",
      "iterations: 6.666666666666667\n",
      "{'DELTA': 1e-06, 'DROPOUT': 0.75, 'MOMENT': 0.999}\n",
      "\n",
      "Train on 6311 samples, validate on 902 samples\n",
      "Epoch 1/7\n",
      "6311/6311 [==============================] - 1s 231us/step - loss: 0.6801 - acc: 0.6097 - val_loss: 0.6665 - val_acc: 0.6231\n",
      "Epoch 2/7\n",
      "6311/6311 [==============================] - 1s 81us/step - loss: 0.6661 - acc: 0.6175 - val_loss: 0.6628 - val_acc: 0.6231\n",
      "Epoch 3/7\n",
      "6311/6311 [==============================] - 1s 81us/step - loss: 0.6657 - acc: 0.6175 - val_loss: 0.6641 - val_acc: 0.6231\n",
      "Epoch 4/7\n",
      "6311/6311 [==============================] - 1s 84us/step - loss: 0.6660 - acc: 0.6175 - val_loss: 0.6627 - val_acc: 0.6231\n",
      "Epoch 5/7\n",
      "6311/6311 [==============================] - 1s 81us/step - loss: 0.6656 - acc: 0.6175 - val_loss: 0.6625 - val_acc: 0.6231\n",
      "Epoch 6/7\n",
      "6311/6311 [==============================] - 0s 79us/step - loss: 0.6656 - acc: 0.6175 - val_loss: 0.6628 - val_acc: 0.6231\n",
      "Epoch 7/7\n",
      "6311/6311 [==============================] - 1s 80us/step - loss: 0.6659 - acc: 0.6175 - val_loss: 0.6625 - val_acc: 0.6231\n",
      "902/902 [==============================] - 0s 73us/step\n",
      "\n",
      "5 seconds.\n",
      "\n",
      "12 | Tue May 29 16:58:21 2018 | lowest loss so far: 0.6625 (run 2)\n",
      "\n",
      "iterations: 6.666666666666667\n",
      "{'DELTA': 0.0001, 'DROPOUT': 0.5, 'MOMENT': 0.9}\n",
      "\n",
      "Train on 6311 samples, validate on 902 samples\n",
      "Epoch 1/7\n",
      "6311/6311 [==============================] - 1s 228us/step - loss: 0.6815 - acc: 0.6119 - val_loss: 0.6626 - val_acc: 0.6231\n",
      "Epoch 2/7\n",
      "6311/6311 [==============================] - 1s 84us/step - loss: 0.6660 - acc: 0.6175 - val_loss: 0.6663 - val_acc: 0.6231\n",
      "Epoch 3/7\n",
      "6311/6311 [==============================] - 1s 80us/step - loss: 0.6658 - acc: 0.6175 - val_loss: 0.6629 - val_acc: 0.6231\n",
      "Epoch 4/7\n",
      "6311/6311 [==============================] - 1s 86us/step - loss: 0.6659 - acc: 0.6175 - val_loss: 0.6650 - val_acc: 0.6231\n",
      "902/902 [==============================] - 0s 82us/step\n",
      "\n",
      "4 seconds.\n",
      "\n",
      "*** 5 configurations x 6.7 iterations each\n",
      "\n",
      "13 | Tue May 29 16:58:24 2018 | lowest loss so far: 0.6625 (run 2)\n",
      "\n",
      "iterations: 6.666666666666666\n",
      "{'DELTA': 0.0001, 'DROPOUT': 0.1, 'MOMENT': 0.9}\n",
      "\n",
      "Train on 6311 samples, validate on 902 samples\n",
      "Epoch 1/7\n",
      "6311/6311 [==============================] - 2s 239us/step - loss: 0.7497 - acc: 0.6105 - val_loss: 0.6649 - val_acc: 0.6231\n",
      "Epoch 2/7\n",
      "6311/6311 [==============================] - 1s 81us/step - loss: 0.6657 - acc: 0.6175 - val_loss: 0.6732 - val_acc: 0.6231\n",
      "Epoch 3/7\n",
      "6311/6311 [==============================] - 1s 82us/step - loss: 0.6627 - acc: 0.6175 - val_loss: 0.6652 - val_acc: 0.6231\n",
      "Epoch 4/7\n",
      "6311/6311 [==============================] - 1s 82us/step - loss: 0.6598 - acc: 0.6175 - val_loss: 0.6936 - val_acc: 0.5166\n",
      "902/902 [==============================] - 0s 79us/step\n",
      "\n",
      "4 seconds.\n",
      "\n",
      "14 | Tue May 29 16:58:28 2018 | lowest loss so far: 0.6625 (run 2)\n",
      "\n",
      "iterations: 6.666666666666666\n",
      "{'DELTA': 1e-06, 'DROPOUT': 0.75, 'MOMENT': 0.99}\n",
      "\n",
      "Train on 6311 samples, validate on 902 samples\n",
      "Epoch 1/7\n",
      "6311/6311 [==============================] - 2s 247us/step - loss: 0.6770 - acc: 0.6124 - val_loss: 0.6689 - val_acc: 0.6231\n",
      "Epoch 2/7\n",
      "6311/6311 [==============================] - 1s 81us/step - loss: 0.6672 - acc: 0.6175 - val_loss: 0.6627 - val_acc: 0.6231\n",
      "Epoch 3/7\n",
      "6311/6311 [==============================] - 1s 81us/step - loss: 0.6650 - acc: 0.6175 - val_loss: 0.6627 - val_acc: 0.6231\n",
      "Epoch 4/7\n",
      "6311/6311 [==============================] - 0s 78us/step - loss: 0.6652 - acc: 0.6175 - val_loss: 0.6659 - val_acc: 0.6231\n",
      "Epoch 5/7\n",
      "6311/6311 [==============================] - 1s 82us/step - loss: 0.6651 - acc: 0.6175 - val_loss: 0.6626 - val_acc: 0.6231\n",
      "Epoch 6/7\n",
      "6311/6311 [==============================] - 1s 83us/step - loss: 0.6634 - acc: 0.6177 - val_loss: 0.6732 - val_acc: 0.6231\n",
      "Epoch 7/7\n",
      "6311/6311 [==============================] - 1s 82us/step - loss: 0.6629 - acc: 0.6178 - val_loss: 0.6653 - val_acc: 0.6231\n",
      "902/902 [==============================] - 0s 74us/step\n",
      "\n",
      "6 seconds.\n",
      "\n",
      "15 | Tue May 29 16:58:34 2018 | lowest loss so far: 0.6625 (run 2)\n",
      "\n",
      "iterations: 6.666666666666666\n",
      "{'DELTA': 0.0001, 'DROPOUT': 0.5, 'MOMENT': 0.999}\n",
      "\n",
      "Train on 6311 samples, validate on 902 samples\n",
      "Epoch 1/7\n",
      "6311/6311 [==============================] - 2s 257us/step - loss: 0.6919 - acc: 0.6099 - val_loss: 0.6663 - val_acc: 0.6231\n",
      "Epoch 2/7\n",
      "6311/6311 [==============================] - 1s 81us/step - loss: 0.6665 - acc: 0.6175 - val_loss: 0.6679 - val_acc: 0.6231\n",
      "Epoch 3/7\n",
      "6311/6311 [==============================] - 1s 86us/step - loss: 0.6670 - acc: 0.6175 - val_loss: 0.6651 - val_acc: 0.6231\n",
      "Epoch 4/7\n",
      "6311/6311 [==============================] - 1s 83us/step - loss: 0.6667 - acc: 0.6175 - val_loss: 0.6691 - val_acc: 0.6231\n",
      "Epoch 5/7\n",
      "6311/6311 [==============================] - 1s 82us/step - loss: 0.6669 - acc: 0.6175 - val_loss: 0.6668 - val_acc: 0.6231\n",
      "Epoch 6/7\n",
      "6311/6311 [==============================] - 1s 82us/step - loss: 0.6661 - acc: 0.6175 - val_loss: 0.6682 - val_acc: 0.6231\n",
      "902/902 [==============================] - 0s 79us/step\n",
      "\n",
      "5 seconds.\n",
      "\n",
      "16 | Tue May 29 16:58:39 2018 | lowest loss so far: 0.6625 (run 2)\n",
      "\n",
      "iterations: 6.666666666666666\n",
      "{'DELTA': 0.0001, 'DROPOUT': 0.5, 'MOMENT': 0.99}\n",
      "\n",
      "Train on 6311 samples, validate on 902 samples\n",
      "Epoch 1/7\n",
      "6311/6311 [==============================] - 2s 266us/step - loss: 0.7020 - acc: 0.6058 - val_loss: 0.6626 - val_acc: 0.6231\n",
      "Epoch 2/7\n",
      "6311/6311 [==============================] - 1s 81us/step - loss: 0.6659 - acc: 0.6175 - val_loss: 0.6629 - val_acc: 0.6231\n",
      "Epoch 3/7\n",
      "6311/6311 [==============================] - 1s 81us/step - loss: 0.6652 - acc: 0.6175 - val_loss: 0.6653 - val_acc: 0.6231\n",
      "Epoch 4/7\n",
      "6311/6311 [==============================] - 1s 84us/step - loss: 0.6662 - acc: 0.6175 - val_loss: 0.6633 - val_acc: 0.6231\n",
      "902/902 [==============================] - 0s 81us/step\n",
      "\n",
      "4 seconds.\n",
      "\n",
      "17 | Tue May 29 16:58:43 2018 | lowest loss so far: 0.6625 (run 2)\n",
      "\n",
      "iterations: 6.666666666666666\n",
      "{'DELTA': 1e-06, 'DROPOUT': 0.1, 'MOMENT': 0.999}\n",
      "\n",
      "Train on 6311 samples, validate on 902 samples\n",
      "Epoch 1/7\n",
      "6311/6311 [==============================] - 2s 277us/step - loss: 0.7182 - acc: 0.6088 - val_loss: 0.6627 - val_acc: 0.6231\n",
      "Epoch 2/7\n",
      "6311/6311 [==============================] - 1s 83us/step - loss: 0.6655 - acc: 0.6175 - val_loss: 0.6648 - val_acc: 0.6231\n",
      "Epoch 3/7\n",
      "6311/6311 [==============================] - 1s 79us/step - loss: 0.6646 - acc: 0.6175 - val_loss: 0.6628 - val_acc: 0.6231\n",
      "Epoch 4/7\n",
      "6311/6311 [==============================] - 1s 84us/step - loss: 0.6631 - acc: 0.6175 - val_loss: 0.6785 - val_acc: 0.6231\n",
      "902/902 [==============================] - 0s 81us/step\n",
      "\n",
      "4 seconds.\n",
      "{'DELTA': 1e-06, 'DROPOUT': 0.75, 'MOMENT': 0.999}\n",
      "Reading /src/myhome/code/sequence-localization/data/050_run_cdna_keras_genomics/cdna_expt01/train.h5.batch1 ...\n",
      "Reading /src/myhome/code/sequence-localization/data/050_run_cdna_keras_genomics/cdna_expt01/train.h5.batch2 ...\n",
      "Reading /src/myhome/code/sequence-localization/data/050_run_cdna_keras_genomics/cdna_expt01/valid.h5.batch1 ...\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 6311 samples, validate on 902 samples\n",
      "Epoch 1/20\n",
      "6311/6311 [==============================] - 2s 301us/step - loss: 0.6901 - categorical_accuracy: 0.6083 - val_loss: 0.6626 - val_categorical_accuracy: 0.6231\n",
      "\n",
      "Epoch 00001: val_loss improved from inf to 0.66264, saving model to /src/myhome/code/sequence-localization/data/050_run_cdna_keras_genomics/cdna_expt01/model/model_bestmodel_weights.h5\n",
      "Epoch 2/20\n",
      "6311/6311 [==============================] - 1s 83us/step - loss: 0.6656 - categorical_accuracy: 0.6175 - val_loss: 0.6636 - val_categorical_accuracy: 0.6231\n",
      "\n",
      "Epoch 00002: val_loss did not improve from 0.66264\n",
      "Epoch 3/20\n",
      "6311/6311 [==============================] - 1s 83us/step - loss: 0.6655 - categorical_accuracy: 0.6175 - val_loss: 0.6637 - val_categorical_accuracy: 0.6231\n",
      "\n",
      "Epoch 00003: val_loss did not improve from 0.66264\n",
      "Epoch 4/20\n",
      "6311/6311 [==============================] - 1s 82us/step - loss: 0.6656 - categorical_accuracy: 0.6175 - val_loss: 0.6646 - val_categorical_accuracy: 0.6231\n",
      "\n",
      "Epoch 00004: val_loss did not improve from 0.66264\n",
      "Epoch 5/20\n",
      "6311/6311 [==============================] - 1s 82us/step - loss: 0.6658 - categorical_accuracy: 0.6175 - val_loss: 0.6637 - val_categorical_accuracy: 0.6231\n",
      "\n",
      "Epoch 00005: val_loss did not improve from 0.66264\n",
      "Epoch 6/20\n",
      "6311/6311 [==============================] - 1s 81us/step - loss: 0.6657 - categorical_accuracy: 0.6175 - val_loss: 0.6625 - val_categorical_accuracy: 0.6231\n",
      "\n",
      "Epoch 00006: val_loss improved from 0.66264 to 0.66255, saving model to /src/myhome/code/sequence-localization/data/050_run_cdna_keras_genomics/cdna_expt01/model/model_bestmodel_weights.h5\n",
      "Epoch 7/20\n",
      "6311/6311 [==============================] - 1s 82us/step - loss: 0.6656 - categorical_accuracy: 0.6175 - val_loss: 0.6625 - val_categorical_accuracy: 0.6231\n",
      "\n",
      "Epoch 00007: val_loss did not improve from 0.66255\n",
      "Epoch 8/20\n",
      "6311/6311 [==============================] - 1s 81us/step - loss: 0.6657 - categorical_accuracy: 0.6175 - val_loss: 0.6625 - val_categorical_accuracy: 0.6231\n",
      "\n",
      "Epoch 00008: val_loss improved from 0.66255 to 0.66255, saving model to /src/myhome/code/sequence-localization/data/050_run_cdna_keras_genomics/cdna_expt01/model/model_bestmodel_weights.h5\n",
      "Epoch 9/20\n",
      "6311/6311 [==============================] - 1s 82us/step - loss: 0.6655 - categorical_accuracy: 0.6175 - val_loss: 0.6627 - val_categorical_accuracy: 0.6231\n",
      "\n",
      "Epoch 00009: val_loss did not improve from 0.66255\n",
      "Epoch 10/20\n",
      "6311/6311 [==============================] - 1s 80us/step - loss: 0.6655 - categorical_accuracy: 0.6175 - val_loss: 0.6626 - val_categorical_accuracy: 0.6231\n",
      "\n",
      "Epoch 00010: val_loss did not improve from 0.66255\n",
      "Epoch 11/20\n",
      "6311/6311 [==============================] - 1s 86us/step - loss: 0.6656 - categorical_accuracy: 0.6175 - val_loss: 0.6626 - val_categorical_accuracy: 0.6231\n",
      "\n",
      "Epoch 00011: val_loss did not improve from 0.66255\n",
      "Epoch 12/20\n",
      "6311/6311 [==============================] - 1s 83us/step - loss: 0.6655 - categorical_accuracy: 0.6175 - val_loss: 0.6626 - val_categorical_accuracy: 0.6231\n",
      "\n",
      "Epoch 00012: val_loss did not improve from 0.66255\n",
      "Epoch 13/20\n",
      "6311/6311 [==============================] - 1s 85us/step - loss: 0.6656 - categorical_accuracy: 0.6175 - val_loss: 0.6628 - val_categorical_accuracy: 0.6231\n",
      "\n",
      "Epoch 00013: val_loss did not improve from 0.66255\n",
      "Epoch 14/20\n",
      "6311/6311 [==============================] - 1s 83us/step - loss: 0.6655 - categorical_accuracy: 0.6175 - val_loss: 0.6629 - val_categorical_accuracy: 0.6231\n",
      "\n",
      "Epoch 00014: val_loss did not improve from 0.66255\n",
      "Epoch 15/20\n",
      "6311/6311 [==============================] - 1s 83us/step - loss: 0.6655 - categorical_accuracy: 0.6175 - val_loss: 0.6626 - val_categorical_accuracy: 0.6231\n",
      "\n",
      "Epoch 00015: val_loss did not improve from 0.66255\n",
      "Epoch 16/20\n",
      "6311/6311 [==============================] - 1s 86us/step - loss: 0.6656 - categorical_accuracy: 0.6175 - val_loss: 0.6627 - val_categorical_accuracy: 0.6231\n",
      "\n",
      "Epoch 00016: val_loss did not improve from 0.66255\n",
      "Epoch 17/20\n",
      "6311/6311 [==============================] - 1s 82us/step - loss: 0.6655 - categorical_accuracy: 0.6175 - val_loss: 0.6630 - val_categorical_accuracy: 0.6231\n",
      "\n",
      "Epoch 00017: val_loss did not improve from 0.66255\n",
      "Epoch 18/20\n",
      "6311/6311 [==============================] - 1s 85us/step - loss: 0.6655 - categorical_accuracy: 0.6175 - val_loss: 0.6631 - val_categorical_accuracy: 0.6231\n",
      "\n",
      "Epoch 00018: val_loss did not improve from 0.66255\n",
      "Test AUC for output neuron 0: 0.5\n",
      "Test categorical accuracy: 0.6084303937881309\n"
     ]
    }
   ],
   "source": [
    "! python /src/myhome/code/Keras-genomics/main.py -d $data_folder/cdna_expt01 -m /src/myhome/code/Keras-genomics/example/model.py -y -t -e"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "# %run -d -b /src/myhome/code/Keras-genomics/hyperband.py:136 /src/myhome/code/Keras-genomics/main.py -d $data_folder/cdna_expt01 -m /src/myhome/code/Keras-genomics/example/model.py -y -t -e"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "test.h5.batch1  train.h5.batch1  train.txt        valid.txt\r\n",
      "test.txt        train.h5.batch2  valid.h5.batch1\r\n"
     ]
    }
   ],
   "source": [
    "ls $data_folder/cdna_expt01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "==> /src/myhome/code/sequence-localization/data/052_run_cdna_keras_genomics_pyrimidine_purine/cdna_expt01/test.txt <==\r\n",
      "/data/cdna_expt01/test.h5.batch1\r\n",
      "\r\n",
      "==> /src/myhome/code/sequence-localization/data/052_run_cdna_keras_genomics_pyrimidine_purine/cdna_expt01/train.txt <==\r\n",
      "/data/cdna_expt01/train.h5.batch1\r\n",
      "/data/cdna_expt01/train.h5.batch2\r\n",
      "\r\n",
      "==> /src/myhome/code/sequence-localization/data/052_run_cdna_keras_genomics_pyrimidine_purine/cdna_expt01/valid.txt <==\r\n",
      "/data/cdna_expt01/valid.h5.batch1\r\n"
     ]
    }
   ],
   "source": [
    "! head $data_folder/cdna_expt01/*txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ls: cannot access 'ls': No such file or directory\r\n",
      "ls: cannot access '/src/myhome/code/sequence-localization/data/052_run_cdna_keras_genomics_pyrimidine_purine/cdna_expt01/model': No such file or directory\r\n"
     ]
    }
   ],
   "source": [
    "! ls ls $data_folder/cdna_expt01/model"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ls: cannot access '/src/myhome/code/sequence-localization/data/052_run_cdna_keras_genomics_pyrimidine_purine/cdna_expt01/model/model.training_history.txt': No such file or directory\r\n"
     ]
    },
    {
     "ename": "FileNotFoundError",
     "evalue": "File b'/src/myhome/code/sequence-localization/data/052_run_cdna_keras_genomics_pyrimidine_purine/cdna_expt01/model/model.training_history.txt' does not exist",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-12-441e74cf4e4a>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mtxt\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpath\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata_folder\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'cdna_expt01'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'model'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'model.training_history.txt'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0mget_ipython\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msystem\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m' ls $txt'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 7\u001b[0;31m \u001b[0mtraining_history\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_table\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtxt\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      8\u001b[0m \u001b[0mtraining_history\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtraining_history\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstrip\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'# '\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtraining_history\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36mparser_f\u001b[0;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, escapechar, comment, encoding, dialect, tupleize_cols, error_bad_lines, warn_bad_lines, skipfooter, doublequote, delim_whitespace, low_memory, memory_map, float_precision)\u001b[0m\n\u001b[1;32m    676\u001b[0m                     skip_blank_lines=skip_blank_lines)\n\u001b[1;32m    677\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 678\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0m_read\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    679\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    680\u001b[0m     \u001b[0mparser_f\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__name__\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_read\u001b[0;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[1;32m    438\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    439\u001b[0m     \u001b[0;31m# Create the parser.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 440\u001b[0;31m     \u001b[0mparser\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mTextFileReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfilepath_or_buffer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    441\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    442\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mchunksize\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0miterator\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[1;32m    785\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'has_index_names'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkwds\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'has_index_names'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    786\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 787\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mengine\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    788\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    789\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mclose\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m_make_engine\u001b[0;34m(self, engine)\u001b[0m\n\u001b[1;32m   1012\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_make_engine\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mengine\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'c'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1013\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'c'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1014\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mCParserWrapper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0moptions\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1015\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1016\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mengine\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'python'\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/conda/lib/python3.6/site-packages/pandas/io/parsers.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, src, **kwds)\u001b[0m\n\u001b[1;32m   1706\u001b[0m         \u001b[0mkwds\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'usecols'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0musecols\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1707\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1708\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reader\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mparsers\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mTextReader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msrc\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1709\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1710\u001b[0m         \u001b[0mpassed_names\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnames\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader.__cinit__\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/parsers.pyx\u001b[0m in \u001b[0;36mpandas._libs.parsers.TextReader._setup_parser_source\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: File b'/src/myhome/code/sequence-localization/data/052_run_cdna_keras_genomics_pyrimidine_purine/cdna_expt01/model/model.training_history.txt' does not exist"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "\n",
    "txt = os.path.join(data_folder, 'cdna_expt01', 'model', 'model.training_history.txt')\n",
    "! ls $txt\n",
    "training_history = pd.read_table(txt)\n",
    "training_history.columns = training_history.columns.str.strip('# ')\n",
    "print(training_history.shape)\n",
    "training_history.head()\n",
    "\n",
    "training_history_reset_index = training_history.reset_index()\n",
    "training_history_reset_index = training_history_reset_index.rename(columns={'index': 'epoch'})\n",
    "training_history_reset_index.head()\n",
    "\n",
    "training_history_tidy = pd.melt(training_history_reset_index, id_vars='epoch')\n",
    "print(training_history_tidy.shape)\n",
    "training_history_tidy.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "g = sns.FacetGrid(data=training_history_tidy, col='variable')\n",
    "g.map(sns.regplot, 'epoch', 'value')\n",
    "g.set(ylim=(0, 1), xlim=(0, 20))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
